In [239]:
import pandas as pd
import numpy as np 
import re
import csv
from tqdm import tqdm
import time

In [3]:
nacoesunidas = pd.read_csv(r'C:\Users\carlo\OneDrive\ANALYTICS\ANALYTICS - IRONHACK\PORTFOLIO\PROJETOS\PROJETO_FINAL_CURSO\Nacoes_Unidas\Dados_Nacoes_Unidas\IAEG+DF_UNDATA_MDG+1.2_2023_05_15_18_02_14.csv', 
                        sep=';',
                        low_memory=False)

In [4]:
nacoesunidas.head()

DATAFLOW FREQ       SERIES     UNIT LOCATION AGE_GROUP  SEX   
0  IAEG:DF_UNDATA_MDG(1.2)    A  AG_LND_FRST  PERCENT        T       NaN  NaN  \
1  IAEG:DF_UNDATA_MDG(1.2)    A  AG_LND_FRST  PERCENT        T       NaN  NaN   
2  IAEG:DF_UNDATA_MDG(1.2)    A  AG_LND_FRST  PERCENT        T       NaN  NaN   
3  IAEG:DF_UNDATA_MDG(1.2)    A  AG_LND_FRST  PERCENT        T       NaN  NaN   
4  IAEG:DF_UNDATA_MDG(1.2)    A  AG_LND_FRST  PERCENT        T       NaN  NaN   

  REF_AREA  SOURCE_TYPE  TIME_PERIOD  OBS_VALUE  UNIT_MULT  TIME_DETAIL   
0      ABW          NaN         1990   0.000000          0         1990  \
1      ABW          NaN         2000   0.000000          0         2000   
2      ABW          NaN         2005   0.000000          0         2005   
3      ABW          NaN         2010   0.000000          0         2010   
4      AFG          NaN         1990   2.070266          0         1990   

  NATURE  SOURCE_DETAIL FOOTNOTES  
0      E            NaN       NaN  
1      E            NaN       NaN  
2      E            NaN       NaN  
3      E            NaN       NaN  
4      E            NaN       NaN

In [5]:
nacoesunidas.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 387044 entries, 0 to 387043
Data columns (total 16 columns):
 #   Column         Non-Null Count   Dtype  
---  ------         --------------   -----  
 0   DATAFLOW       387044 non-null  object 
 1   FREQ           387044 non-null  object 
 2   SERIES         387044 non-null  object 
 3   UNIT           387044 non-null  object 
 4   LOCATION       387044 non-null  object 
 5   AGE_GROUP      304147 non-null  object 
 6   SEX            302994 non-null  object 
 7   REF_AREA       387044 non-null  object 
 8   SOURCE_TYPE    0 non-null       float64
 9   TIME_PERIOD    387044 non-null  int64  
 10  OBS_VALUE      320677 non-null  float64
 11  UNIT_MULT      387044 non-null  int64  
 12  TIME_DETAIL    387044 non-null  int64  
 13  NATURE         315236 non-null  object 
 14  SOURCE_DETAIL  0 non-null       float64
 15  FOOTNOTES      94713 non-null   object 
dtypes: float64(3), int64(3), object(10)
memory usage: 47.2+ MB


In [52]:
nacoesunidas['REF_AREA'] = nacoesunidas['REF_AREA'].replace({  'CAN': 'Canada',
  'CPV': 'Cape Verde',
  'CYM': 'Cayman Islands',
  'CAF': 'Central African Republic',
  'TCD': 'Chad',
  '830': 'Channel Islands',
  'CHL': 'Chile',
  'CHN': 'China',
  'HKG': 'Hong Kong Special Administrative Region of China',
  'MAC': 'Macao Special Administrative Region of China',
  'COL': 'Colombia',
  'COM': 'Comoros',
  'COG': 'Democratic Republic of the Congo',
  'COK': 'Cook Islands',
  'CRI': 'Costa Rica',
  'CIV': "Côte d'Ivoire",
  'HRV': 'Croatia',
  'CUB': 'Cuba',
  'CYP': 'Cyprus',
  'CZE': 'Czech Republic',
  'PRK': "Democratic People's Republic of Korea",
  'COD': 'Democratic Republic of the Congo',
  'DNK': 'Denmark',
  'DJI': 'Djibouti',
  'DMA': 'Dominica',
  'DOM': 'Dominican Republic',
  'ECU': 'Ecuador',
  'EGY': 'Egypt',
  'SLV': 'El Salvador',
  'GNQ': 'Equatorial Guinea',
  'ERI': 'Eritrea',
  'EST': 'Estonia',
  'ETH': 'Ethiopia',
  'FRO': 'Faeroe Islands',
  'FLK': 'Falkland Islands (Malvinas)',
  'FJI': 'Fiji',
  'FIN': 'Finland',
  'FRA': 'France',
  'GUF': 'French Guiana',
  'PYF': 'French Polynesia',
  'GAB': 'Gabon',
  'GMB': 'Gambia',
  'GEO': 'Georgia',
  'DEU': 'Germany',
  'GHA': 'Ghana',
  'GIB': 'Gibraltar',
  'GRC': 'Greece',
  'GRL': 'Greenland',
  'GRD': 'Grenada',
  'GLP': 'Guadeloupe',
  'GUM': 'Guam',
  'GTM': 'Guatemala',
  'GIN': 'Guinea',
  'GNB': 'Guinea-Bissau',
  'GUY': 'Guyana',
  'HTI': 'Haiti',
  'HND': 'Honduras',
  'HUN': 'Hungary',
  'ISL': 'Iceland',
  'IND': 'India',
  'IDN': 'Indonesia',
  'IRN': 'Iran (Islamic Republic of)',
  'IRQ': 'Iraq',
  'IRL': 'Ireland',
  'ISR': 'Israel',
  'ITA': 'Italy',
  'JAM': 'Jamaica',
  'AFG': 'Afghanistan',
  'ALB': 'Albania',
  'DZA': 'Algeria',
  'ASM': 'American Samoa',
  'AND': 'Andorra',
  'AGO': 'Angola',
  'AIA': 'Anguilla',
  'ATG': 'Antigua and Barbuda',
  'ARG': 'Argentina',
  'ARM': 'Armenia',
  'ABW': 'Aruba',
  'AUS': 'Australia',
  'AUT': 'Austria',
  'AZE': 'Azerbaijan',
  'BHS': 'Bahamas',
  'BHR': 'Bahrain',
  'BGD': 'Bangladesh',
  'BRB': 'Barbados',
  'BLR': 'Belarus',
  'BEL': 'Belgium',
  'BLZ': 'Belize',
  'BEN': 'Benin',
  'BMU': 'Bermuda',
  'BTN': 'Bhutan',
  'BOL': 'Bolivia (Plurinational State of)',
  'BIH': 'Bosnia and Herzegovina',
  'BWA': 'Botswana',
  'BRA': 'Brazil',
  'VGB': 'British Virgin Islands',
  'BRN': 'Brunei Darussalam',
  'BGR': 'Bulgaria',
  'BFA': 'Burkina Faso',
  'BDI': 'Burundi',
  'KHM': 'Cambodia',
  'CMR': 'Cameroon',
  'JPN': 'Japan',
  'JOR': 'Jordan',
  'KAZ': 'Kazakhstan',
  'KEN': 'Kenya',
  'KIR': 'Kiribati',
  'KWT': 'Kuwait',
  'KGZ': 'Kyrgyzstan',
  'LAO': "Lao People's Democratic Republic",
  'LVA': 'Latvia',
  'LBN': 'Lebanon',
  'LSO': 'Lesotho',
  'LBR': 'Liberia',
  'LBY': 'Libyan Arab Jamahiriya',
  'LIE': 'Liechtenstein',
  'LTU': 'Lithuania',
  'LUX': 'Luxembourg',
  'MDG': 'Madagascar',
  'MWI': 'Malawi',
  'MYS': 'Malaysia',
  'MDV': 'Maldives',
  'MLI': 'Mali',
  'MLT': 'Malta',
  'MHL': 'Marshall Islands',
  'MTQ': 'Martinique',
  'MRT': 'Mauritania',
  'MUS': 'Mauritius',
  'MYT': 'Mayotte',
  'MEX': 'Mexico',
  'FSM': 'Micronesia (Federated States of)',
  'MDA': 'Republic of Moldova',
  'MCO': 'Monaco',
  'MNG': 'Mongolia',
  'MNE': 'Montenegro',
  'MSR': 'Montserrat',
  'MAR': 'Morocco',
  'MOZ': 'Mozambique',
  'MMR': 'Myanmar',
  'NAM': 'Namibia',
  'NRU': 'Nauru',
  'NPL': 'Nepal',
  'NLD': 'Netherlands',
  'ANT': 'Netherlands Antilles',
  'NCL': 'New Caledonia',
  'NZL': 'New Zealand',
  'NIC': 'Nicaragua',
  'NER': 'Niger',
  'NGA': 'Nigeria',
  'NIU': 'Niue',
  'MNP': 'Northern Mariana Islands',
  'NOR': 'Norway',
  'PSE': 'State of Palestine',
  'OMN': 'Oman',
  'PAK': 'Pakistan',
  'PLW': 'Palau',
  'PAN': 'Panama',
  'PNG': 'Papua New Guinea',
  'PRY': 'Paraguay',
  'PER': 'Peru',
  'PHL': 'Philippines',
  'POL': 'Poland',
  'PRT': 'Portugal',
  'PRI': 'Puerto Rico',
  'QAT': 'Qatar',
  'KOR': 'Republic of Korea',
  'REU': 'Réunion',
  'ROU': 'Romania',
  'RUS': 'Russian Federation',
  'RWA': 'Rwanda',
  'SHN': 'Saint Helena',
  'KNA': 'Saint Kitts and Nevis',
  'LCA': 'Saint Lucia',
  'SPM': 'Saint Pierre and Miquelon',
  'VCT': 'Saint Vincent and the Grenadines',
  'WSM': 'Samoa',
  'SMR': 'San Marino',
  'STP': 'Sao Tome and Principe',
  'SAU': 'Saudi Arabia',
  'SEN': 'Senegal',
  'SRB': 'Serbia',
  'SYC': 'Seychelles',
  'SLE': 'Sierra Leone',
  'SGP': 'Singapore',
  'SVK': 'Slovakia',
  'SVN': 'Slovenia',
  'SLB': 'Solomon Islands',
  'SOM': 'Somalia',
  'ZAF': 'South Africa',
  'SSD': 'South Sudan',
  'ESP': 'Spain',
  'LKA': 'Sri Lanka',
  'SDN': 'Sudan',
  'SUR': 'Suriname',
  'SWZ': 'Swaziland',
  'SWE': 'Sweden',
  'CHE': 'Switzerland',
  'SYR': 'Syrian Arab Republic',
  'TJK': 'Tajikistan',
  'THA': 'Thailand',
  'MKD': 'The former Yugoslav Republic of Macedonia',
  'TLS': 'Timor-Leste',
  'TGO': 'Togo',
  'TKL': 'Tokelau',
  'TON': 'Tonga',
  'TTO': 'Trinidad and Tobago',
  'TUN': 'Tunisia',
  'TUR': 'Turkey',
  'TKM': 'Turkmenistan',
  'TCA': 'Turks and Caicos Islands',
  'TUV': 'Tuvalu',
  'UGA': 'Uganda',
  'UKR': 'Ukraine',
  'ARE': 'United Arab Emirates',
  'GBR': 'United Kingdom of Great Britain and Northern Ireland',
  'TZA': 'United Republic of Tanzania',
  'USA': 'United States of America',
  'VIR': 'United States Virgin Islands',
  'URY': 'Uruguay',
  'UZB': 'Uzbekistan',
  'VUT': 'Vanuatu',
  'VEN': 'Venezuela (Bolivarian Republic of)',
  'VNM': 'Viet Nam',
  'WLF': 'Wallis and Futuna Islands',
  'ESH': 'Western Sahara',
  'YEM': 'Yemen',
  'ZMB': 'Zambia',
  'ZWE': 'Zimbabwe'})

In [53]:
nacoesunidas.head()

DATAFLOW FREQ                       SERIES     UNIT   
0  IAEG:DF_UNDATA_MDG(1.2)    A  Land area covered by forest  PERCENT  \
1  IAEG:DF_UNDATA_MDG(1.2)    A  Land area covered by forest  PERCENT   
2  IAEG:DF_UNDATA_MDG(1.2)    A  Land area covered by forest  PERCENT   
3  IAEG:DF_UNDATA_MDG(1.2)    A  Land area covered by forest  PERCENT   
4  IAEG:DF_UNDATA_MDG(1.2)    A  Land area covered by forest  PERCENT   

  LOCATION AGE_GROUP  SEX     REF_AREA  SOURCE_TYPE  TIME_PERIOD  OBS_VALUE   
0        T       NaN  NaN        Aruba          NaN         1990   0.000000  \
1        T       NaN  NaN        Aruba          NaN         2000   0.000000   
2        T       NaN  NaN        Aruba          NaN         2005   0.000000   
3        T       NaN  NaN        Aruba          NaN         2010   0.000000   
4        T       NaN  NaN  Afghanistan          NaN         1990   2.070266   

   UNIT_MULT  TIME_DETAIL NATURE  SOURCE_DETAIL FOOTNOTES  
0          0         1990      E            NaN       NaN  
1          0         2000      E            NaN       NaN  
2          0         2005      E            NaN       NaN  
3          0         2010      E            NaN       NaN  
4          0         1990      E            NaN       NaN

In [54]:
df_series = "[AG_LND_FRST] - Land area covered by forest[DC_AID_LLDC] - ODA received in landlocked developing countries[DC_AID_LLDCG] - ODA received in landlocked developing countries as a percentage of their GNI[DC_AID_SIDS] - ODA received in small island developing states[DC_AID_SIDSG] - ODA received in small island developing states as a percentage of their GNI[DC_HPC_COMR] - Debt relief committed under HIPC Initiative, cumulative[DC_HPC_MDRI] - Debt relief delivered in full under MDRI initiative, cumulative[DC_HPC_STTS] - Countries that have reached their HIPC decision points and countries that have reached their HIPC completion points (cumulative)[DC_ODA_LDCG] - Net ODA to LDCs as a percentage of OECD/DAC donors' GNI[DC_ODA_LDCS] - Net ODA to LDCs[DC_ODA_SOCL] - ODA to basic social services[DC_ODA_SOCP] - ODA to basic social services as a percentage of sector-allocable ODA[DC_ODA_TCAP] - ODA provided to help build trade capacity[DC_ODA_TOTG] - Net ODA as a percentage of OECD/DAC donors' GNI[DC_ODA_TOTL] - Net ODA[DC_ODA_UNTD] - ODA that is untied[DT_TDS_DECT] - Debt service as percentage of exports of goods and services and net income[EG_USE_COMM] - Energy use[EG_USE_SLFL] - Population using solid fuels[EN_ATM_CFC] - Consumption of ozone-depleting CFCs[EN_ATM_CO2] - Carbon dioxide emissions[EN_ATM_CO2_CDIAC] - Carbon dioxide emissions (CDIAC)[EN_ATM_CO2PC] - Carbon dioxide emissions per capita[EN_ATM_CO2PC_CDIAC] - Carbon dioxide emissions per capita (CDIAC)[EN_ATM_ODS] - Consumption of all ozone-depleting substances[EN_LND_SLUM] - Urban population living in slums[ER_LND_TMPA] - Terrestrial and marine areas protected[ER_LND_TPA] - Terrestrial areas protected[ER_MRN_MPA] - Marine areas protected[IT_SUB_i911] - Mobile cellular telephone subscriptions[IT_SUB_ii91] - Telephone lines[IT_USE_ii99] - Internet users[NR_WTR_WITH] - Total water resources used[PA_NUS_PPP] - Purchasing power parities (PPP) conversion factor[SE_ADT_1524] - Literacy rate[SE_ADT_GPI] - Women to men parity index, as ratio of literacy rates[SE_ENR_NORPH] - School attendance rate of children both of whose parents are alive and who live with at least one parent[SE_ENR_ORPH] - School attendance rate of orphans[SE_ENR_ORPHR] - Ratio of school attendance rate of orphans to school attendance rate of non-orphans[SE_PRM_CMPL] - Primary completion rate[SE_PRM_GPI] - Gender Parity Index in primary level enrolment[SE_PRM_NENR] - Total net enrolment ratio in primary education[SE_PRM_PRSL] - Pupils starting grade 1 who reach last grade of primary[SE_SEC_GPI] - Gender Parity Index in secondary level enrolment[SE_TER_GPI] - Gender Parity Index in tertiary level enrolment[SG_GEN_PARL] - Seats in national parliament[SH_ARV_ACCS] - Antiretroviral therapy coverage among people with advanced HIV infection[SH_ARV_ACCS_LB] - Antiretroviral therapy coverage among people with advanced HIV infection (lower bound)[SH_ARV_ACCS_UB] - Antiretroviral therapy coverage among people with advanced HIV infection (upper bound)[SH_ARV_MTCT] - HIV-infected pregnant women who received antiretroviral drugs to reduce the risk for mother-to-child transmission[SH_ARV_MTCT_LB] - HIV-infected pregnant women who received antiretroviral drugs to reduce the risk for mother-to-child transmission (lower bound)[SH_ARV_MTCT_UB] - HIV-infected pregnant women who received antiretroviral drugs to reduce the risk for mother-to-child transmission (upper bound)[SH_CON_HRSX] - Condom use at last high-risk sex[SH_DYN_AIDS] - AIDS deaths[SH_DYN_AIDS_LB] - AIDS deaths (lower bound)[SH_DYN_AIDS_UB] - AIDS deaths (upper bound)[SH_DYN_IMRT] - Infant mortality rate[SH_DYN_MORT] - Under-five mortality rate[SH_FPL_UNML] - Unmet need for family planning, limiting[SH_FPL_UNMS] - Unmet need for family planning, spacing[SH_FPL_UNMT] - Unmet need for family planning[SH_H2O_IMPR] - Population using improved drinking water sources[SH_HIV_INCD] - HIV incidence rate[SH_HIV_INCD_LB] - HIV incidence rate (lower bound)[SH_HIV_INCD_UB] - HIV incidence rate (upper bound)[SH_HIV_KNOW] - Population with comprehensive correct knowledge of HIV/AIDS[SH_HIV_ORPH] - AIDS orphans[SH_HIV_TOTL] - People living with HIV[SH_HIV_TOTL_LB] - People living with HIV (lower bound)[SH_HIV_TOTL_UB] - People living with HIV (upper bound)[SH_IMM_MEAS] - Children immunized against measles[SH_MLR_MORT] - Malaria death rate[SH_MLR_NETS] - Children sleeping under insecticide-treated bed nets[SH_MLR_TRET] - Children with fever being treated with anti-malarial drugs[SH_SAN_IMPR] - Population using improved sanitation facilities[SH_STA_ANV1] - Antenatal care coverage, at least one visit[SH_STA_ANV4] - Antenatal care coverage, at least four visits[SH_STA_BRTC] - Births attended by skilled health personnel[SH_STA_MALR] - Notified cases of malaria[SH_STA_MORT] - Maternal mortality ratio[SH_TBS_CURE] - Tuberculosis treatment success rate under DOTS[SH_TBS_DOTS] - Tuberculosis detection rate under DOTS[SH_TBS_DOTS_LB] - Tuberculosis detection rate under DOTS (lower bound)[SH_TBS_DOTS_UB] - Tuberculosis detection rate under DOTS (upper bound)[SH_TBS_INCD] - Tuberculosis incidence rate[SH_TBS_INCD_LB] - Tuberculosis incidence rate (lower bound)[SH_TBS_INCD_UB] - Tuberculosis incidence rate (upper bound)[SH_TBS_MORT] - Tuberculosis death rate[SH_TBS_MORT_LB] - Tuberculosis death rate (lower bound)[SH_TBS_MORT_UB] - Tuberculosis death rate (upper bound)[SH_TBS_PREV] - Tuberculosis prevalence rate[SH_TBS_PREV_LB] - Tuberculosis prevalence rate (lower bound)[SH_TBS_PREV_UB] - Tuberculosis prevalence rate (upper bound)[SI_DST_1QNT] - Poorest quintile's share in national income or consumption[SI_POV_DAY1] - Population below USD 1 (PPP) per day[SI_POV_GAP1] - Poverty gap ratio at USD 1 a day (PPP)[SI_POV_NAHC] - Population below national poverty line[SL_EMP_1DAY] - Employed people living below USD 1 (PPP) per day[SL_EMP_NAGR] - Share women in wage employment in the non-agricultural sector[SL_EMP_OACF] - Proportion of own-account and contributing family workers in total employment[SL_EMP_PCAP] - Growth rate of GDP per person employed[SL_EMP_TOTL] - Employment-to-population ratio[SL_UEM_1524] - Youth unemployment rate[SL_UEM_YTOA] - Ratio of youth unemployment rate to adult unemployment rate[SL_UEM_YTOT] - Share of youth unemployed in total unemployed[SL_UEM_YTOY] - Share of youth unemployed in youth population[SN_ITK_DEFC] - Population undernourished[SN_STA_MALN] - Children moderately or severely underweight[SN_STA_MALS] - Children severely underweight[SP_DYN_ADKL] - Adolescent birth rate[SP_DYN_CCON] - Condom use to overall contraceptive use among currently married women[SP_DYN_CONC] - Contraceptive use among married women, condom[SP_DYN_CONM] - Contraceptive use among married women, modern methods[SP_DYN_CONU] - Contraceptive use among married women, any method[TM_VAL_DCDF] - Developed country imports from developing countries, admitted duty free[TM_VAL_LDDF] - Developed country imports from the LDCs, admitted duty free[TT_ASE_TOTG] - Agricultural support estimate for OECD countries as a percentage of their GDP[TT_ASE_TOTL] - Agricultural support estimate for OECD countries"

In [55]:
padrao = re.sub('\[','\n', df_series)
padrao = re.sub('\]',':', padrao)
padrao = re.sub('\-','', padrao)
padrao = re.sub(':',"':'", padrao)
padrao = re.sub('\n',"',\n'", padrao)
padrao = re.sub("'  ","'", padrao)

In [56]:
print(padrao)

',
'AG_LND_FRST':'Land area covered by forest',
'DC_AID_LLDC':'ODA received in landlocked developing countries',
'DC_AID_LLDCG':'ODA received in landlocked developing countries as a percentage of their GNI',
'DC_AID_SIDS':'ODA received in small island developing states',
'DC_AID_SIDSG':'ODA received in small island developing states as a percentage of their GNI',
'DC_HPC_COMR':'Debt relief committed under HIPC Initiative, cumulative',
'DC_HPC_MDRI':'Debt relief delivered in full under MDRI initiative, cumulative',
'DC_HPC_STTS':'Countries that have reached their HIPC decision points and countries that have reached their HIPC completion points (cumulative)',
'DC_ODA_LDCG':'Net ODA to LDCs as a percentage of OECD/DAC donors' GNI',
'DC_ODA_LDCS':'Net ODA to LDCs',
'DC_ODA_SOCL':'ODA to basic social services',
'DC_ODA_SOCP':'ODA to basic social services as a percentage of sectorallocable ODA',
'DC_ODA_TCAP':'ODA provided to help build trade capacity',
'DC_ODA_TOTG':'Net ODA as a percentage

In [57]:
nacoesunidas['SERIES'] = nacoesunidas['SERIES'].replace({'AG_LND_FRST':'Land area covered by forest',
'DC_AID_LLDC':'ODA received in landlocked developing countries',
'DC_AID_LLDCG':'ODA received in landlocked developing countries as a percentage of their GNI',
'DC_AID_SIDS':'ODA received in small island developing states',
'DC_AID_SIDSG':'ODA received in small island developing states as a percentage of their GNI',
'DC_HPC_COMR':'Debt relief committed under HIPC Initiative, cumulative',
'DC_HPC_MDRI':'Debt relief delivered in full under MDRI initiative, cumulative',
'DC_HPC_STTS':'Countries that have reached their HIPC decision points and countries that have reached their HIPC completion points (cumulative)',
'DC_ODA_LDCG':"Net ODA to LDCs as a percentage of OECD/DAC donors' GNI",
'DC_ODA_LDCS':'Net ODA to LDCs',
'DC_ODA_SOCL':'ODA to basic social services',
'DC_ODA_SOCP':'ODA to basic social services as a percentage of sectorallocable ODA',
'DC_ODA_TCAP':'ODA provided to help build trade capacity',
'DC_ODA_TOTG':"Net ODA as a percentage of OECD/DAC donors' GNI",
'DC_ODA_TOTL':'Net ODA',
'DC_ODA_UNTD':'ODA that is untied',
'DT_TDS_DECT':'Debt service as percentage of exports of goods and services and net income',
'EG_USE_COMM':'Energy use',
'EG_USE_SLFL':'Population using solid fuels',
'EN_ATM_CFC':'Consumption of ozonedepleting CFCs',
'EN_ATM_CO2':'Carbon dioxide emissions',
'EN_ATM_CO2_CDIAC':'Carbon dioxide emissions (CDIAC)',
'EN_ATM_CO2PC':'Carbon dioxide emissions per capita',
'EN_ATM_CO2PC_CDIAC':'Carbon dioxide emissions per capita (CDIAC)',
'EN_ATM_ODS':'Consumption of all ozonedepleting substances',
'EN_LND_SLUM':'Urban population living in slums',
'ER_LND_TMPA':'Terrestrial and marine areas protected',
'ER_LND_TPA':'Terrestrial areas protected',
'ER_MRN_MPA':'Marine areas protected',
'IT_SUB_i911':'Mobile cellular telephone subscriptions',
'IT_SUB_ii91':'Telephone lines',
'IT_USE_ii99':'Internet users',
'NR_WTR_WITH':'Total water resources used',
'PA_NUS_PPP':'Purchasing power parities (PPP) conversion factor',
'SE_ADT_1524':'Literacy rate',
'SE_ADT_GPI':'Women to men parity index, as ratio of literacy rates',
'SE_ENR_NORPH':'School attendance rate of children both of whose parents are alive and who live with at least one parent',
'SE_ENR_ORPH':'School attendance rate of orphans',
'SE_ENR_ORPHR':'Ratio of school attendance rate of orphans to school attendance rate of nonorphans',
'SE_PRM_CMPL':'Primary completion rate',
'SE_PRM_GPI':'Gender Parity Index in primary level enrolment',
'SE_PRM_NENR':'Total net enrolment ratio in primary education',
'SE_PRM_PRSL':'Pupils starting grade 1 who reach last grade of primary',
'SE_SEC_GPI':'Gender Parity Index in secondary level enrolment',
'SE_TER_GPI':'Gender Parity Index in tertiary level enrolment',
'SG_GEN_PARL':'Seats in national parliament',
'SH_ARV_ACCS':'Antiretroviral therapy coverage among people with advanced HIV infection',
'SH_ARV_ACCS_LB':'Antiretroviral therapy coverage among people with advanced HIV infection (lower bound)',
'SH_ARV_ACCS_UB':'Antiretroviral therapy coverage among people with advanced HIV infection (upper bound)',
'SH_ARV_MTCT':'HIVinfected pregnant women who received antiretroviral drugs to reduce the risk for mothertochild transmission',
'SH_ARV_MTCT_LB':'HIVinfected pregnant women who received antiretroviral drugs to reduce the risk for mothertochild transmission (lower bound)',
'SH_ARV_MTCT_UB':'HIVinfected pregnant women who received antiretroviral drugs to reduce the risk for mothertochild transmission (upper bound)',
'SH_CON_HRSX':'Condom use at last highrisk sex',
'SH_DYN_AIDS':'AIDS deaths',
'SH_DYN_AIDS_LB':'AIDS deaths (lower bound)',
'SH_DYN_AIDS_UB':'AIDS deaths (upper bound)',
'SH_DYN_IMRT':'Infant mortality rate',
'SH_DYN_MORT':'Underfive mortality rate',
'SH_FPL_UNML':'Unmet need for family planning, limiting',
'SH_FPL_UNMS':'Unmet need for family planning, spacing',
'SH_FPL_UNMT':'Unmet need for family planning',
'SH_H2O_IMPR':'Population using improved drinking water sources',
'SH_HIV_INCD':'HIV incidence rate',
'SH_HIV_INCD_LB':'HIV incidence rate (lower bound)',
'SH_HIV_INCD_UB':'HIV incidence rate (upper bound)',
'SH_HIV_KNOW':'Population with comprehensive correct knowledge of HIV/AIDS',
'SH_HIV_ORPH':'AIDS orphans',
'SH_HIV_TOTL':'People living with HIV',
'SH_HIV_TOTL_LB':'People living with HIV (lower bound)',
'SH_HIV_TOTL_UB':'People living with HIV (upper bound)',
'SH_IMM_MEAS':'Children immunized against measles',
'SH_MLR_MORT':'Malaria death rate',
'SH_MLR_NETS':'Children sleeping under insecticidetreated bed nets',
'SH_MLR_TRET':'Children with fever being treated with antimalarial drugs',
'SH_SAN_IMPR':'Population using improved sanitation facilities',
'SH_STA_ANV1':'Antenatal care coverage, at least one visit',
'SH_STA_ANV4':'Antenatal care coverage, at least four visits',
'SH_STA_BRTC':'Births attended by skilled health personnel',
'SH_STA_MALR':'Notified cases of malaria',
'SH_STA_MORT':'Maternal mortality ratio',
'SH_TBS_CURE':'Tuberculosis treatment success rate under DOTS',
'SH_TBS_DOTS':'Tuberculosis detection rate under DOTS',
'SH_TBS_DOTS_LB':'Tuberculosis detection rate under DOTS (lower bound)',
'SH_TBS_DOTS_UB':'Tuberculosis detection rate under DOTS (upper bound)',
'SH_TBS_INCD':'Tuberculosis incidence rate',
'SH_TBS_INCD_LB':'Tuberculosis incidence rate (lower bound)',
'SH_TBS_INCD_UB':'Tuberculosis incidence rate (upper bound)',
'SH_TBS_MORT':'Tuberculosis death rate',
'SH_TBS_MORT_LB':'Tuberculosis death rate (lower bound)',
'SH_TBS_MORT_UB':'Tuberculosis death rate (upper bound)',
'SH_TBS_PREV':'Tuberculosis prevalence rate',
'SH_TBS_PREV_LB':'Tuberculosis prevalence rate (lower bound)',
'SH_TBS_PREV_UB':'Tuberculosis prevalence rate (upper bound)',
'SI_DST_1QNT':"Poorest quintile's share in national income or consumption",
'SI_POV_DAY1':'Population below USD 1 (PPP) per day',
'SI_POV_GAP1':'Poverty gap ratio at USD 1 a day (PPP)',
'SI_POV_NAHC':'Population below national poverty line',
'SL_EMP_1DAY':'Employed people living below USD 1 (PPP) per day',
'SL_EMP_NAGR':'Share women in wage employment in the nonagricultural sector',
'SL_EMP_OACF':'Proportion of ownaccount and contributing family workers in total employment',
'SL_EMP_PCAP':'Growth rate of GDP per person employed',
'SL_EMP_TOTL':'Employmenttopopulation ratio',
'SL_UEM_1524':'Youth unemployment rate',
'SL_UEM_YTOA':'Ratio of youth unemployment rate to adult unemployment rate',
'SL_UEM_YTOT':'Share of youth unemployed in total unemployed',
'SL_UEM_YTOY':'Share of youth unemployed in youth population',
'SN_ITK_DEFC':'Population undernourished',
'SN_STA_MALN':'Children moderately or severely underweight',
'SN_STA_MALS':'Children severely underweight',
'SP_DYN_ADKL':'Adolescent birth rate',
'SP_DYN_CCON':'Condom use to overall contraceptive use among currently married women',
'SP_DYN_CONC':'Contraceptive use among married women, condom',
'SP_DYN_CONM':'Contraceptive use among married women, modern methods',
'SP_DYN_CONU':'Contraceptive use among married women, any method',
'TM_VAL_DCDF':'Developed country imports from developing countries, admitted duty free',
'TM_VAL_LDDF':'Developed country imports from the LDCs, admitted duty free',
'TT_ASE_TOTG':'Agricultural support estimate for OECD countries as a percentage of their GDP',
'TT_ASE_TOTL':'Agricultural support estimate for OECD countries'})

In [58]:
nacoesunidas

DATAFLOW FREQ   
0       IAEG:DF_UNDATA_MDG(1.2)    A  \
1       IAEG:DF_UNDATA_MDG(1.2)    A   
2       IAEG:DF_UNDATA_MDG(1.2)    A   
3       IAEG:DF_UNDATA_MDG(1.2)    A   
4       IAEG:DF_UNDATA_MDG(1.2)    A   
...                         ...  ...   
387039  IAEG:DF_UNDATA_MDG(1.2)    A   
387040  IAEG:DF_UNDATA_MDG(1.2)    A   
387041  IAEG:DF_UNDATA_MDG(1.2)    A   
387042  IAEG:DF_UNDATA_MDG(1.2)    A   
387043  IAEG:DF_UNDATA_MDG(1.2)    A   

                                                  SERIES     UNIT LOCATION   
0                            Land area covered by forest  PERCENT        T  \
1                            Land area covered by forest  PERCENT        T   
2                            Land area covered by forest  PERCENT        T   
3                            Land area covered by forest  PERCENT        T   
4                            Land area covered by forest  PERCENT        T   
...                                                  ...      ...      ...   
387039  Agricultural support estimate for OECD countries      USD        T   
387040  Agricultural support estimate for OECD countries      USD        T   
387041  Agricultural support estimate for OECD countries      USD        T   
387042  Agricultural support estimate for OECD countries      USD        T   
387043  Agricultural support estimate for OECD countries      USD        T   

       AGE_GROUP  SEX                  REF_AREA  SOURCE_TYPE  TIME_PERIOD   
0            NaN  NaN                     Aruba          NaN         1990  \
1            NaN  NaN                     Aruba          NaN         2000   
2            NaN  NaN                     Aruba          NaN         2005   
3            NaN  NaN                     Aruba          NaN         2010   
4            NaN  NaN               Afghanistan          NaN         1990   
...          ...  ...                       ...          ...          ...   
387039       NaN  NaN  United States of America          NaN         2009   
387040       NaN  NaN  United States of America          NaN         2010   
387041       NaN  NaN  United States of America          NaN         2011   
387042       NaN  NaN  United States of America          NaN         2012   
387043       NaN  NaN  United States of America          NaN         2013   

           OBS_VALUE  UNIT_MULT  TIME_DETAIL NATURE  SOURCE_DETAIL   
0           0.000000          0         1990      E            NaN  \
1           0.000000          0         2000      E            NaN   
2           0.000000          0         2005      E            NaN   
3           0.000000          0         2010      E            NaN   
4           2.070266          0         1990      E            NaN   
...              ...        ...          ...    ...            ...   
387039  76164.930000          6         2009      C            NaN   
387040  76904.280000          6         2010      C            NaN   
387041  74963.310000          6         2011      C            NaN   
387042  82048.830000          6         2012      C            NaN   
387043  83084.310000          6         2013      C            NaN   

                                                FOOTNOTES  
0                                                     NaN  
1                                                     NaN  
2                                                     NaN  
3                                                     NaN  
4                                                     NaN  
...                                                   ...  
387039               Based on OECD, PSE/CSE database 2015  
387040               Based on OECD, PSE/CSE database 2015  
387041               Based on OECD, PSE/CSE database 2015  
387042               Based on OECD, PSE/CSE database 2015  
387043  Provisional.| Based on OECD, PSE/CSE database ...  

[387044 rows x 16 columns]

In [59]:
nacoesunidas.columns

Index(['DATAFLOW', 'FREQ', 'SERIES', 'UNIT', 'LOCATION', 'AGE_GROUP', 'SEX',
       'REF_AREA', 'SOURCE_TYPE', 'TIME_PERIOD', 'OBS_VALUE', 'UNIT_MULT',
       'TIME_DETAIL', 'NATURE', 'SOURCE_DETAIL', 'FOOTNOTES'],
      dtype='object')

In [60]:
nacoesunidas['UNIT'].unique()

array(['PERCENT', 'USD', 'PV_USD_2009', 'NUMBER',
       'KG_OE_PER_GDP_CON_PPP_USD', 'T', 'KG_CO2_PER_GDP_CON_PPP_USD',
       'KM2', 'PER_100_POP', 'LCU_PPP_USD', 'RATIO',
       'PER_1000_LIVE_BIRTHS', 'PER_100000_POP', 'PER_100000_LIVE_BIRTHS',
       'PER_1000_POP'], dtype=object)

In [61]:
nacoes_poverty = nacoesunidas.loc[nacoesunidas['SERIES'] == 'Poverty gap ratio at USD 1 a day (PPP)']

In [62]:
nacoes_poverty.isna().sum()

DATAFLOW           0
FREQ               0
SERIES             0
UNIT               0
LOCATION           0
AGE_GROUP          0
SEX                0
REF_AREA           0
SOURCE_TYPE      852
TIME_PERIOD        0
OBS_VALUE          0
UNIT_MULT          0
TIME_DETAIL        0
NATURE             0
SOURCE_DETAIL    852
FOOTNOTES          1
dtype: int64

In [63]:
nacoes_poverty = nacoes_poverty[['SERIES','REF_AREA','UNIT','OBS_VALUE','TIME_PERIOD']]

In [64]:
nacoes_poverty.info()

<class 'pandas.core.frame.DataFrame'>
Index: 852 entries, 320295 to 321146
Data columns (total 5 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   SERIES       852 non-null    object 
 1   REF_AREA     852 non-null    object 
 2   UNIT         852 non-null    object 
 3   OBS_VALUE    852 non-null    float64
 4   TIME_PERIOD  852 non-null    int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 39.9+ KB


In [65]:
nacoes_poverty = nacoes_poverty.rename(columns={'SERIES':'INDICADOR','REF_AREA':'PAIS','UNIT':'UNIDADE','OBS_VALUE':'VALOR','TIME_PERIOD':'ANO'})

In [66]:
nacoes_poverty.head()

INDICADOR     PAIS  UNIDADE  VALOR   ANO
320295  Poverty gap ratio at USD 1 a day (PPP)   Angola  PERCENT  29.94  2000
320296  Poverty gap ratio at USD 1 a day (PPP)   Angola  PERCENT  16.45  2008
320297  Poverty gap ratio at USD 1 a day (PPP)  Albania  PERCENT   0.02  1996
320298  Poverty gap ratio at USD 1 a day (PPP)  Albania  PERCENT   0.12  2002
320299  Poverty gap ratio at USD 1 a day (PPP)  Albania  PERCENT   0.11  2004

In [67]:
# nacoes_poverty.to_excel(r'C:\Users\carlo\OneDrive\ANALYTICS\ANALYTICS - IRONHACK\PORTFOLIO\PROJETOS\PROJETO_FINAL_CURSO\Nacoes_Unidas\Dados_Nacoes_Unidas\Nacoes_Poverty.xlsx', index=False)

In [68]:
len(nacoes_poverty['PAIS'].unique())

124

In [69]:
len(nacoesunidas['REF_AREA'].unique())

228

In [70]:
nacoesunidas_2 = pd.read_csv(r'C:\Users\carlo\OneDrive\ANALYTICS\ANALYTICS - IRONHACK\PORTFOLIO\PROJETOS\PROJETO_FINAL_CURSO\Nacoes_Unidas\Dados_Nacoes_Unidas\COLUNAS_SEPARADAS_2_IAEG-SDGs+DF_SDG_GLH+1.13_2023_05_18_01_13_13.csv', 
                        sep=';',
                        low_memory=False)

In [71]:
nacoesunidas_2.head()

INDICADOR  PAIS   ANO    VALOR UNIDADE
0  BX_KLT_DINV_WD_GD_ZS     1  1970  0.47921      PT
1  BX_KLT_DINV_WD_GD_ZS     1  1971   0.4702      PT
2  BX_KLT_DINV_WD_GD_ZS     1  1972  0.42351      PT
3  BX_KLT_DINV_WD_GD_ZS     1  1973  0.47988      PT
4  BX_KLT_DINV_WD_GD_ZS     1  1974  0.49684      PT

In [72]:
nacoesunidas_2['INDICADOR'] = nacoesunidas_2['INDICADOR'].replace({'SI_POV_NAHC':'Proportion of population living below the national poverty line [1.2.1]',
'SI_COV_UEMP':'Proportion of unemployed persons receiving unemployment cash benefit [1.3.1]',
'SI_COV_VULN':'Proportion of vulnerable population receiving social assistance cash benefit [1.3.1]',
'SP_ACS_BSRVSAN':'Proportion of population using basic sanitation services [1.4.1]',
'SP_ACS_BSRVH2O':'Proportion of population using basic drinking water services [1.4.1]',
'SN_ITK_DEFC':'Prevalence of undernourishment [2.1.1]',
'SN_ITK_DEFCN':'Number of undernourish people [2.1.1]',
'SH_STA_BRTC':'Proportion of births attended by skilled health personnel [3.1.2]',
'SH_DYN_IMRT':'Infant mortality rate [3.2.1]',
'SH_DTH_NCOM':'Mortality rate attributed to cardiovascular disease, cancer, diabetes or chronic respiratory disease [3.4.1]',
'SH_ALC_CONSPT':'Alcohol consumption per capita (aged 15 years and older) within a calendar year [3.5.2]',
'SH_STA_WASHARI':'Mortality rate attributed to unsafe water, unsafe sanitation and lack of hygiene from diarrhoea, intestinal nematode infections, malnutrition and acute respiratory infections [3.9.2]',
'SH_MED_DEN':'Health worker density, by type of occupation (per 10,000 population) [3.c.1]',
'SH_MED_HWRKDIS':'Health worker distribution, by sex and type of occupation [3.c.1]',
'SE_ADT_EDUCTRN':'Participation rate in formal and non-formal education and training [4.3.1]',
'SE_ADT_ACTS':'Proportion of youth and adults with information and communications technology (ICT) skills [4.4.1]',
'SE_GPI_PART':'Adjusted gender parity index for participation rate in formal and non-formal education and training (ratio) [4.5.1]',
'SE_GPI_ICTS':'Gender parity index for youth/adults with information and communications technology (ICT) skills [4.5.1]',
'VC_VAW_MARR':'Proportion of ever-partnered women and girls subjected to physical and sexual violence by a current or former intimate partner in the previous 12 months [5.2.1]',
'SG_GEN_LOCGELS':'Proportion of elected seats held by women in deliberative bodies of local government [5.5.1]',
'SG_GEN_PARL':'Proportion of seats held by women in national parliaments (% of total number of seats) [5.5.1]',
'SG_GEN_PARLN':'Number of seats held by women in national parliaments [5.5.1]',
'SG_GEN_PARLNT':'Current number of seats in national parliaments [5.5.1]',
'IT_MOB_OWN':'Proportion of individuals who own a mobile telephone [5.b.1]',
'EG_ACS_ELEC':'Proportion of population with access to electricity [7.1.1]',
'EG_FEC_RNEW':'Renewable energy share in the total final energy consumption [7.2.1]',
'NY_GDP_PCAP':'Annual growth rate of real GDP per capita [8.1.1]',
'SL_EMP_PCAP':'Annual growth rate of real GDP per employed person [8.2.1]',
'SL_EMP_EARN':'Average hourly earnings [8.5.1]',
'SL_TLF_UEM':'Unemployment rate, by sex and age - 13th ICLS (%) [8.5.2]',
'BX_KLT_DINV_WD_GD_ZS':'Foreign direct investment, net inflows, as a proportion of GDP (%) [17.13.1]',
'NY_GDP_MKTP_KD_ZG':'Annual GDP growth (%) [17.13.1]',
'FP_CPI_TOTL_ZG':'Annual inflation, consumer prices (%) [17.13.1]',
'TG_VAL_TOTL_GD_ZS':'Merchandise trade as a proportion of GDP (%) [17.13.1]'})

In [73]:
nacoesunidas_2.head()

INDICADOR  PAIS   ANO    VALOR   
0  Foreign direct investment, net inflows, as a p...     1  1970  0.47921  \
1  Foreign direct investment, net inflows, as a p...     1  1971   0.4702   
2  Foreign direct investment, net inflows, as a p...     1  1972  0.42351   
3  Foreign direct investment, net inflows, as a p...     1  1973  0.47988   
4  Foreign direct investment, net inflows, as a p...     1  1974  0.49684   

  UNIDADE  
0      PT  
1      PT  
2      PT  
3      PT  
4      PT

In [74]:
nacoesunidas_2['INDICADOR'].unique()

array(['Foreign direct investment, net inflows, as a proportion of GDP (%) [17.13.1]',
       'Renewable energy share in the total final energy consumption [7.2.1]',
       'Annual inflation, consumer prices (%) [17.13.1]',
       'Proportion of individuals who own a mobile telephone [5.b.1]',
       'Annual GDP growth (%) [17.13.1]',
       'Annual growth rate of real GDP per capita [8.1.1]',
       'SE_ACS_ELECT',
       'Proportion of youth and adults with information and communications technology (ICT) skills [4.4.1]',
       'Participation rate in formal and non-formal education and training [4.3.1]',
       'Gender parity index for youth/adults with information and communications technology (ICT) skills [4.5.1]',
       'Adjusted gender parity index for participation rate in formal and non-formal education and training (ratio) [4.5.1]',
       'Proportion of elected seats held by women in deliberative bodies of local government [5.5.1]',
       'Proportion of seats held by women 

In [75]:
nacoesunidas_2['INDICADOR'] = nacoesunidas_2['INDICADOR'].str.replace(r'\[.*\]', '', regex=True).str.strip()
nacoesunidas_2['INDICADOR'].unique()

array(['Foreign direct investment, net inflows, as a proportion of GDP (%)',
       'Renewable energy share in the total final energy consumption',
       'Annual inflation, consumer prices (%)',
       'Proportion of individuals who own a mobile telephone',
       'Annual GDP growth (%)',
       'Annual growth rate of real GDP per capita', 'SE_ACS_ELECT',
       'Proportion of youth and adults with information and communications technology (ICT) skills',
       'Participation rate in formal and non-formal education and training',
       'Gender parity index for youth/adults with information and communications technology (ICT) skills',
       'Adjusted gender parity index for participation rate in formal and non-formal education and training (ratio)',
       'Proportion of elected seats held by women in deliberative bodies of local government',
       'Proportion of seats held by women in national parliaments (% of total number of seats)',
       'Number of seats held by women in natio

In [76]:
pais_prov = pd.read_csv(r'C:\Users\carlo\OneDrive\ANALYTICS\ANALYTICS - IRONHACK\PORTFOLIO\PROJETOS\PROJETO_FINAL_CURSO\Nacoes_Unidas\Dados_Nacoes_Unidas\paises_segundo_df.csv')

In [77]:
pais_prov

Empty DataFrame
Columns: [<select onchange="results.events.change(this)" id="SKV_REF_AREA" title="[1] - World" style="width: 394.667px	><option value=1" selected="selected" title="[1] - World">[1] - World</option><option value="100" title="[100] - Bulgaria">[100] - Bulgaria</option><option value="104" title="[104] - Myanmar">[104] - Myanmar</option><option value="108" title="[108] - Burundi">[108] - Burundi</option><option value="11" title="[11] - Western Africa">[11] - Western Africa</option><option value="112" title="[112] - Belarus">[112] - Belarus</option><option value="116" title="[116] - Cambodia">[116] - Cambodia</option><option value="12" title="[12] - Algeria">[12] - Algeria</option><option value="120" title="[120] - Cameroon">[120] - Cameroon</option><option value="124" title="[124] - Canada">[124] - Canada</option><option value="13" title="[13] - Central America">[13] - Central America</option><option value="132" title="[132] - Cabo Verde">[132] - Cabo Verde</option><option value="136" title="[136] - Cayman Islands">[136] - Cayman Islands</option><option value="14" title="[14] - Eastern Africa">[14] - Eastern Africa</option><option value="140" title="[140] - Central African Republic">[140] - Central African Republic</option><option value="142" title="[142] - Asia">[142] - Asia</option><option value="143" title="[143] - Central Asia">[143] - Central Asia</option><option value="144" title="[144] - Sri Lanka">[144] - Sri Lanka</option><option value="145" title="[145] - Western Asia">[145] - Western Asia</option><option value="148" title="[148] - Chad">[148] - Chad</option><option value="15" title="[15] - Northern Africa">[15] - Northern Africa</option><option value="150" title="[150] - Europe">[150] - Europe</option><option value="151" title="[151] - Eastern Europe">[151] - Eastern Europe</option><option value="152" title="[152] - Chile">[152] - Chile</option><option value="154" title="[154] - Northern Europe">[154] - Northern Europe</option><option value="155" title="[155] - Western Europe">[155] - Western Europe</option><option value="156" title="[156] - China">[156] - China</option><option value="17" title="[17] - Middle Africa">[17] - Middle Africa</option><option value="170" title="[170] - Colombia">[170] - Colombia</option><option value="174" title="[174] - Comoros">[174] - Comoros</option><option value="178" title="[178] - Congo">[178] - Congo</option><option value="18" title="[18] - Southern Africa">[18] - Southern Africa</option><option value="180" title="[180] - Democratic Republic of the Congo">[180] - Democratic Republic of the Congo</option><option value="188" title="[188] - Costa Rica">[188] - Costa Rica</option><option value="19" title="[19] - Americas">[19] - Americas</option><option value="191" title="[191] - Croatia">[191] - Croatia</option><option value="196" title="[196] - Cyprus">[196] - Cyprus</option><option value="199" title="[199] - Least Developed Countries (LDCs)">[199] - Least Developed Countries (LDCs)</option><option value="2" title="[2] - Africa">[2] - Africa</option><option value="202" title="[202] - Sub-Saharan Africa">[202] - Sub-Saharan Africa</option><option value="203" title="[203] - Czechia">[203] - Czechia</option><option value="204" title="[204] - Benin">[204] - Benin</option><option value="208" title="[208] - Denmark">[208] - Denmark</option><option value="21" title="[21] - Northern America">[21] - Northern America</option><option value="212" title="[212] - Dominica">[212] - Dominica</option><option value="214" title="[214] - Dominican Republic">[214] - Dominican Republic</option><option value="218" title="[218] - Ecuador">[218] - Ecuador</option><option value="222" title="[222] - El Salvador">[222] - El Salvador</option><option value="226" title="[226] - Equatorial Guinea">[226] - Equatorial Guinea</option><option value="231" title="[231] - Ethiopia">[231] - Ethiopia</option><option value="232" title="[232] - Eritrea">[232] - Eritrea</option><option value="233" title="[233] - E

In [79]:
pattern = r'\[(\d+)\]\s-\s([a-zA-Z\s]+)'  # Regex para extrair [código] e nome do país para substituir no dataframe
matches = re.findall(pattern, str(pais_prov))  # Encontra todos os padrões no texto
results = [(match[0]+ ':' + match[1]) for match in matches]  # Cria uma lista de tuplas com os valores correspondentes aos padrões encontrados

nome_pais = []  # Cria uma lista vazia chamada 'nome_pais'

for result in results:
    nome_pais.append(result)  # Adiciona cada tupla de resultado à lista 'nome_pais'
    '\n'

print(nome_pais)

['1:World', '1:World', '1:World', '100:Bulgaria', '100:Bulgaria', '104:Myanmar', '104:Myanmar', '108:Burundi', '108:Burundi', '11:Western Africa', '11:Western Africa', '112:Belarus', '112:Belarus', '116:Cambodia', '116:Cambodia', '12:Algeria', '12:Algeria', '120:Cameroon', '120:Cameroon', '124:Canada', '124:Canada', '13:Central America', '13:Central America', '132:Cabo Verde', '132:Cabo Verde', '136:Cayman Islands', '136:Cayman Islands', '14:Eastern Africa', '14:Eastern Africa', '140:Central African Republic', '140:Central African Republic', '142:Asia', '142:Asia', '143:Central Asia', '143:Central Asia', '144:Sri Lanka', '144:Sri Lanka', '145:Western Asia', '145:Western Asia', '148:Chad', '148:Chad', '15:Northern Africa', '15:Northern Africa', '150:Europe', '150:Europe', '151:Eastern Europe', '151:Eastern Europe', '152:Chile', '152:Chile', '154:Northern Europe', '154:Northern Europe', '155:Western Europe', '155:Western Europe', '156:China', '156:China', '17:Middle Africa', '17:Middle A

In [80]:
# Substituir os valores da coluna 'PAIS' pelos nomes dos países
nacoesunidas_2['PAIS'] = nacoesunidas_2['PAIS'].replace({1:'World',1:'World',1:'World',100:'Bulgaria',100:'Bulgaria',104:'Myanmar',104:'Myanmar',108:'Burundi',108:'Burundi',11:'Western Africa',11:'Western Africa',112:'Belarus',112:'Belarus',116:'Cambodia',116:'Cambodia',12:'Algeria',12:'Algeria',120:'Cameroon',120:'Cameroon',124:'Canada',124:'Canada',13:'Central America',13:'Central America',132:'Cabo Verde',132:'Cabo Verde',136:'Cayman Islands',136:'Cayman Islands',14:'Eastern Africa',14:'Eastern Africa',140:'Central African Republic',140:'Central African Republic',142:'Asia',142:'Asia',143:'Central Asia',143:'Central Asia',144:'Sri Lanka',144:'Sri Lanka',145:'Western Asia',145:'Western Asia',148:'Chad',148:'Chad',15:'Northern Africa',15:'Northern Africa',150:'Europe',150:'Europe',151:'Eastern Europe',151:'Eastern Europe',152:'Chile',152:'Chile',154:'Northern Europe',154:'Northern Europe',155:'Western Europe',155:'Western Europe',156:'China',156:'China',17:'Middle Africa',17:'Middle Africa',170:'Colombia',170:'Colombia',174:'Comoros',174:'Comoros',178:'Democratic Republic of the Congo',178:'Democratic Republic of the Congo',18:'Southern Africa',18:'Southern Africa',180:'Democratic Republic of the Congo',180:'Democratic Republic of the Congo',188:'Costa Rica',188:'Costa Rica',19:'Americas',19:'Americas',191:'Croatia',191:'Croatia',196:'Cyprus',196:'Cyprus',199:'Least Developed Countries ',199:'Least Developed Countries ',2:'Africa',2:'Africa',202:'Sub',202:'Sub',203:'Czechia',203:'Czechia',204:'Benin',204:'Benin',208:'Denmark',208:'Denmark',21:'Northern America',21:'Northern America',212:'Dominica',212:'Dominica',214:'Dominican Republic',214:'Dominican Republic',218:'Ecuador',218:'Ecuador',222:'El Salvador',222:'El Salvador',226:'Equatorial Guinea',226:'Equatorial Guinea',231:'Ethiopia',231:'Ethiopia',232:'Eritrea',232:'Eritrea',233:'Estonia',233:'Estonia',24:'Angola',24:'Angola',242:'Fiji',242:'Fiji',246:'Finland',246:'Finland',250:'France',250:'France',258:'French Polynesia',258:'French Polynesia',262:'Djibouti',262:'Djibouti',266:'Gabon',266:'Gabon',268:'Georgia',268:'Georgia',270:'Gambia',270:'Gambia',275:'State of Palestine',275:'State of Palestine',276:'Germany',276:'Germany',28:'Antigua and Barbuda',28:'Antigua and Barbuda',288:'Ghana',288:'Ghana',296:'Kiribati',296:'Kiribati',30:'Eastern Asia',30:'Eastern Asia',300:'Greece',300:'Greece',308:'Grenada',308:'Grenada',31:'Azerbaijan',31:'Azerbaijan',32:'Argentina',32:'Argentina',320:'Guatemala',320:'Guatemala',324:'Guinea',324:'Guinea',328:'Guyana',328:'Guyana',332:'Haiti',332:'Haiti',34:'Southern Asia',34:'Southern Asia',340:'Honduras',340:'Honduras',344:'China Hong Kong',344:'China Hong Kong',348:'Hungary',348:'Hungary',35:'South',35:'South',352:'Iceland',352:'Iceland',356:'India',356:'India',36:'Australia',36:'Australia',360:'Indonesia',360:'Indonesia',364:'Iran ',364:'Iran ',368:'Iraq',368:'Iraq',372:'Ireland',372:'Ireland',376:'Israel',376:'Israel',380:'Italy',380:'Italy',384:'C',384:'C',388:'Jamaica',388:'Jamaica',39:'Southern Europe',39:'Southern Europe',392:'Japan',392:'Japan',398:'Kazakhstan',398:'Kazakhstan',4:'Afghanistan',4:'Afghanistan',40:'Austria',40:'Austria',400:'Jordan',400:'Jordan',404:'Kenya',404:'Kenya',410:'Republic of Korea',410:'Republic of Korea',412:'Kosovo',412:'Kosovo',414:'Kuwait',414:'Kuwait',417:'Kyrgyzstan',417:'Kyrgyzstan',418:'Lao People',418:'Lao People',419:'Latin America and the Caribbean',419:'Latin America and the Caribbean',422:'Lebanon',422:'Lebanon',426:'Lesotho',426:'Lesotho',428:'Latvia',428:'Latvia',430:'Liberia',430:'Liberia',432:'Landlocked developing countries ',432:'Landlocked developing countries ',434:'Libya',434:'Libya',438:'Liechtenstein',438:'Liechtenstein',44:'Bahamas',44:'Bahamas',440:'Lithuania',440:'Lithuania',442:'Luxembourg',442:'Luxembourg',446:'China Macao',446:'China Macao',450:'Madagascar',450:'Madagascar',454:'Malawi',454:'Malawi',458:'Malaysia',458:'Malaysia',462:'Maldives',462:'Maldives',466:'Mali',466:'Mali',470:'Malta',470:'Malta',478:'Mauritania',478:'Mauritania',48:'Bahrain',48:'Bahrain',480:'Mauritius',480:'Mauritius',484:'Mexico',484:'Mexico',496:'Mongolia',496:'Mongolia',498:'Republic of Moldova',498:'Republic of Moldova',499:'Montenegro',499:'Montenegro',5:'South America',5:'South America',50:'Bangladesh',50:'Bangladesh',504:'Morocco',504:'Morocco',508:'Mozambique',508:'Mozambique',51:'Armenia',51:'Armenia',512:'Oman',512:'Oman',513:'Europe and Northern America',513:'Europe and Northern America',516:'Namibia',516:'Namibia',52:'Barbados',52:'Barbados',524:'Nepal',524:'Nepal',528:'Netherlands ',528:'Netherlands ',53:'Australia and New Zealand',53:'Australia and New Zealand',531:'Cura',531:'Cura',533:'Aruba',533:'Aruba',534:'Sint Maarten ',534:'Sint Maarten ',54:'Melanesia',54:'Melanesia',540:'New Caledonia',540:'New Caledonia',543:'Oceania ',543:'Oceania ',548:'Vanuatu',548:'Vanuatu',554:'New Zealand',554:'New Zealand',558:'Nicaragua',558:'Nicaragua',56:'Belgium',56:'Belgium',562:'Niger',562:'Niger',566:'Nigeria',566:'Nigeria',57:'Micronesia',57:'Micronesia',578:'Norway',578:'Norway',583:'Micronesia ',583:'Micronesia ',584:'Marshall Islands',584:'Marshall Islands',585:'Palau',585:'Palau',586:'Pakistan',586:'Pakistan',591:'Panama',591:'Panama',598:'Papua New Guinea',598:'Papua New Guinea',60:'Bermuda',60:'Bermuda',600:'Paraguay',600:'Paraguay',604:'Peru',604:'Peru',608:'Philippines',608:'Philippines',61:'Polynesia',61:'Polynesia',616:'Poland',616:'Poland',62:'Central and Southern Asia',62:'Central and Southern Asia',620:'Portugal',620:'Portugal',624:'Guinea',624:'Guinea',626:'Timor',626:'Timor',634:'Qatar',634:'Qatar',64:'Bhutan',64:'Bhutan',642:'Romania',642:'Romania',643:'Russian Federation',643:'Russian Federation',646:'Rwanda',646:'Rwanda',659:'Saint Kitts and Nevis',659:'Saint Kitts and Nevis',662:'Saint Lucia',662:'Saint Lucia',670:'Saint Vincent and the Grenadines',670:'Saint Vincent and the Grenadines',678:'Sao Tome and Principe',678:'Sao Tome and Principe',68:'Bolivia ',68:'Bolivia ',682:'Saudi Arabia',682:'Saudi Arabia',686:'Senegal',686:'Senegal',688:'Serbia',688:'Serbia',690:'Seychelles',690:'Seychelles',694:'Sierra Leone',694:'Sierra Leone',70:'Bosnia and Herzegovina',70:'Bosnia and Herzegovina',702:'Singapore',702:'Singapore',703:'Slovakia',703:'Slovakia',704:'Viet Nam',704:'Viet Nam',705:'Slovenia',705:'Slovenia',706:'Somalia',706:'Somalia',710:'South Africa',710:'South Africa',716:'Zimbabwe',716:'Zimbabwe',72:'Botswana',72:'Botswana',722:'Small island developing States ',722:'Small island developing States ',724:'Spain',724:'Spain',728:'South Sudan',728:'South Sudan',729:'Sudan',729:'Sudan',740:'Suriname',740:'Suriname',747:'Northern Africa and Western Asia',747:'Northern Africa and Western Asia',748:'Eswatini',748:'Eswatini',752:'Sweden',752:'Sweden',753:'Eastern and South',753:'Eastern and South',756:'Switzerland',756:'Switzerland',76:'Brazil',76:'Brazil',760:'Syrian Arab Republic',760:'Syrian Arab Republic',762:'Tajikistan',762:'Tajikistan',764:'Thailand',764:'Thailand',768:'Togo',768:'Togo',776:'Tonga',776:'Tonga',780:'Trinidad and Tobago',780:'Trinidad and Tobago',784:'United Arab Emirates',784:'United Arab Emirates',788:'Tunisia',788:'Tunisia',792:'T',792:'T',795:'Turkmenistan',795:'Turkmenistan',796:'Turks and Caicos Islands',796:'Turks and Caicos Islands',798:'Tuvalu',798:'Tuvalu',8:'Albania',8:'Albania',800:'Uganda',800:'Uganda',804:'Ukraine',804:'Ukraine',807:'North Macedonia',807:'North Macedonia',818:'Egypt',818:'Egypt',826:'United Kingdom of Great Britain and Northern Ireland',826:'United Kingdom of Great Britain and Northern Ireland',834:'United Republic of Tanzania',834:'United Republic of Tanzania',84:'Belize',84:'Belize',840:'United States of America',840:'United States of America',854:'Burkina Faso',854:'Burkina Faso',858:'Uruguay',858:'Uruguay',860:'Uzbekistan',860:'Uzbekistan',862:'Venezuela ',862:'Venezuela ',882:'Samoa',882:'Samoa',887:'Yemen',887:'Yemen',894:'Zambia',894:'Zambia',9:'Oceania',9:'Oceania',90:'Solomon Islands',90:'Solomon Islands',96:'Brunei Darussalam',96:'Brunei Darussalam',1:'World',2:'Africa',15:'Northern Africa',12:'Algeria',434:'Libya',504:'Morocco',729:'Sudan',732:'Western Sahara',788:'Tunisia',818:'Egypt',202:'Sub',11:'Western Africa',132:'Cabo Verde',178:'Democratic Republic of the Congo',204:'Benin',270:'Gambia',288:'Ghana',324:'Guinea',384:'C',430:'Liberia',466:'Mali',478:'Mauritania',562:'Niger',566:'Nigeria',624:'Guinea',654:'Saint Helena',686:'Senegal',694:'Sierra Leone',768:'Togo',854:'Burkina Faso',14:'Eastern Africa',86:'British Indian Ocean Territory',108:'Burundi',174:'Comoros',175:'Mayotte',231:'Ethiopia',232:'Eritrea',260:'French Southern Territories',262:'Djibouti',404:'Kenya',450:'Madagascar',454:'Malawi',480:'Mauritius',508:'Mozambique',638:'R',646:'Rwanda',690:'Seychelles',706:'Somalia',716:'Zimbabwe',728:'South Sudan',736:'Sudan ',800:'Uganda',834:'United Republic of Tanzania',894:'Zambia',17:'Middle Africa',24:'Angola',120:'Cameroon',140:'Central African Republic',148:'Chad',180:'Democratic Republic of the Congo',226:'Equatorial Guinea',266:'Gabon',678:'Sao Tome and Principe',18:'Southern Africa',72:'Botswana',426:'Lesotho',516:'Namibia',710:'South Africa',748:'Eswatini',9:'Oceania',53:'Australia and New Zealand',36:'Australia',162:'Christmas Island',166:'Cocos ',334:'Heard Island and McDonald Islands',554:'New Zealand',574:'Norfolk Island',54:'Melanesia',90:'Solomon Islands',242:'Fiji',540:'New Caledonia',548:'Vanuatu',598:'Papua New Guinea',57:'Micronesia',296:'Kiribati',316:'Guam',520:'Nauru',580:'Northern Mariana Islands',581:'United States Minor Outlying Islands',583:'Micronesia ',584:'Marshall Islands',585:'Palau',61:'Polynesia',16:'American Samoa',184:'Cook Islands',258:'French Polynesia',570:'Niue',612:'Pitcairn',772:'Tokelau',776:'Tonga',798:'Tuvalu',876:'Wallis and Futuna Islands',882:'Samoa',543:'Oceania ',10:'Antarctica',19:'Americas',21:'Northern America',60:'Bermuda',124:'Canada',304:'Greenland',666:'Saint Pierre and Miquelon',840:'United States of America',419:'Latin America and the Caribbean',5:'South America',32:'Argentina',68:'Bolivia ',74:'Bouvet Island',76:'Brazil',152:'Chile',170:'Colombia',218:'Ecuador',238:'Falkland Islands ',239:'South Georgia and the South Sandwich Islands',254:'French Guiana',328:'Guyana',600:'Paraguay',604:'Peru',740:'Suriname',858:'Uruguay',862:'Venezuela ',13:'Central America',84:'Belize',188:'Costa Rica',222:'El Salvador',320:'Guatemala',340:'Honduras',484:'Mexico',558:'Nicaragua',591:'Panama',29:'Caribbean',28:'Antigua and Barbuda',44:'Bahamas',52:'Barbados',92:'British Virgin Islands',136:'Cayman Islands',192:'Cuba',212:'Dominica',214:'Dominican Republic',308:'Grenada',312:'Guadeloupe',332:'Haiti',388:'Jamaica',474:'Martinique',500:'Montserrat',530:'Netherlands Antilles',531:'Cura',533:'Aruba',534:'Sint Maarten ',535:'Bonaire',630:'Puerto Rico',652:'Saint Barth',659:'Saint Kitts and Nevis',660:'Anguilla',662:'Saint Lucia',663:'Saint Martin ',670:'Saint Vincent and the Grenadines',780:'Trinidad and Tobago',796:'Turks and Caicos Islands',850:'United States Virgin Islands',62:'Central and Southern Asia',142:'Asia',30:'Eastern Asia',156:'China',158:'Other non',344:'China Hong Kong',392:'Japan',408:'Democratic People',410:'Republic of Korea',446:'China Macao',496:'Mongolia',34:'Southern Asia',4:'Afghanistan',50:'Bangladesh',64:'Bhutan',144:'Sri Lanka',356:'India',364:'Iran ',462:'Maldives',524:'Nepal',586:'Pakistan',35:'South',96:'Brunei Darussalam',104:'Myanmar',116:'Cambodia',360:'Indonesia',418:'Lao People',458:'Malaysia',608:'Philippines',626:'Timor',702:'Singapore',704:'Viet Nam',764:'Thailand',143:'Central Asia',398:'Kazakhstan',417:'Kyrgyzstan',762:'Tajikistan',795:'Turkmenistan',860:'Uzbekistan',145:'Western Asia',31:'Azerbaijan',48:'Bahrain',51:'Armenia',196:'Cyprus',268:'Georgia',275:'State of Palestine',368:'Iraq',369:'Iraq ',370:'Iraq ',376:'Israel',400:'Jordan',414:'Kuwait',422:'Lebanon',512:'Oman',634:'Qatar',682:'Saudi Arabia',760:'Syrian Arab Republic',784:'United Arab Emirates',792:'T',887:'Yemen',223:'Eastern Asia ',150:'Europe',39:'Southern Europe',8:'Albania',20:'Andorra',70:'Bosnia and Herzegovina',191:'Croatia',292:'Gibraltar',300:'Greece',336:'Holy See',380:'Italy',470:'Malta',499:'Montenegro',620:'Portugal',674:'San Marino',688:'Serbia',705:'Slovenia',724:'Spain',807:'North Macedonia',891:'Serbia and Montenegro ',151:'Eastern Europe',100:'Bulgaria',112:'Belarus',203:'Czechia',348:'Hungary',412:'Kosovo',498:'Republic of Moldova',616:'Poland',642:'Romania',643:'Russian Federation',703:'Slovakia',804:'Ukraine',154:'Northern Europe',208:'Denmark',233:'Estonia',234:'Faroe Islands',246:'Finland',352:'Iceland',372:'Ireland',428:'Latvia',440:'Lithuania',578:'Norway',744:'Svalbard and Jan Mayen Islands',752:'Sweden',826:'United Kingdom of Great Britain and Northern Ireland',827:'United Kingdom ',828:'United Kingdom ',829:'United Kingdom ',830:'Channel Islands',680:'Sark',831:'Guernsey',832:'Jersey',833:'Isle of Man',155:'Western Europe',40:'Austria',56:'Belgium',250:'France',276:'Germany',438:'Liechtenstein',442:'Luxembourg',492:'Monaco',528:'Netherlands ',756:'Switzerland',513:'Europe and Northern America',747:'Northern Africa and Western Asia',753:'Eastern and South',127:'Southern Asia ',135:'Caucasus and Central Asia',199:'Least Developed Countries ',420:'Latin America',432:'Landlocked developing countries ',485:'Western Asia ',514:'Developed regions ',515:'Developing regions',518:'Eastern Asia ',593:'Development Assistance Committee members ',722:'Small island developing States ',738:'Sub',746:'Northern Africa ',890:'Yugoslavia ',901:'Africa ',903:'Central and Eastern Europe ',904:'Middle East and North Africa ',905:'Middle East ',911:'Low income economies ',921:'ODA residual',97:'European Union',487:'Other non',527:'Other non',636:'Other non',906:'North America ',917:'International Centers ',918:'European Union ',919:'Regional Centres ',951:'Other Africa ',952:'Unallocated by country',99018:'FAO Major Fishing Area',99019:'FAO Major Fishing Area',99020:'FAO Major Fishing Area',99021:'FAO Major Fishing Area',99022:'FAO Major Fishing Area',99023:'FAO Major Fishing Area',99024:'FAO Major Fishing Area',99025:'FAO Major Fishing Area',99026:'FAO Major Fishing Area',99027:'FAO Major Fishing Area',99028:'FAO Major Fishing Area',99029:'FAO Major Fishing Area',99030:'FAO Major Fishing Area',99031:'FAO Major Fishing Area',99032:'FAO Major Fishing Area',99035:'Residual',99036:'Residual',99037:'Residual',99038:'Residual',99039:'Residual',99040:'Residual',99041:'Belgium and Luxembourg',99049:'Residual',99050:'Residual'})

In [81]:
nacoesunidas_2['PAIS'].unique()

array(['World', 'Africa', 'Afghanistan', 'South America', 'Albania',
       'Oceania', 'Western Africa', 'Algeria', 'Central America',
       'Eastern Africa', 'Northern Africa', 'Middle Africa',
       'Southern Africa', 'Americas', 'Northern America', 'Angola',
       'Antigua and Barbuda', 'Eastern Asia', 'Azerbaijan', 'Argentina',
       'Southern Asia', 'South', 'Australia', 'Southern Europe',
       'Austria', 'Bahamas', 'Bahrain', 'Bangladesh', 'Armenia',
       'Barbados', 'Australia and New Zealand', 'Melanesia', 'Belgium',
       'Micronesia', 'Bermuda', 'Polynesia', 'Central and Southern Asia',
       'Bhutan', 'Bolivia ', 'Bosnia and Herzegovina', 'Botswana',
       'Brazil', 'Belize', 'Solomon Islands', 'Brunei Darussalam',
       'Bulgaria', 'Myanmar', 'Burundi', 'Belarus', 'Cambodia',
       'Cameroon', 'Canada', 'Cabo Verde', 'Cayman Islands',
       'Central African Republic', 'Asia', 'Central Asia', 'Sri Lanka',
       'Western Asia', 'Chad', 'Europe', 'Eastern Europe

In [82]:
nacoesunidas_2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 311680 entries, 0 to 311679
Data columns (total 5 columns):
 #   Column     Non-Null Count   Dtype 
---  ------     --------------   ----- 
 0   INDICADOR  311680 non-null  object
 1   PAIS       311680 non-null  object
 2   ANO        311680 non-null  int64 
 3   VALOR      308673 non-null  object
 4   UNIDADE    311680 non-null  object
dtypes: int64(1), object(4)
memory usage: 11.9+ MB


In [83]:
nacoes_poverty = nacoes_poverty.reindex(columns=nacoesunidas_2.columns)
nacoes_poverty.info()

<class 'pandas.core.frame.DataFrame'>
Index: 852 entries, 320295 to 321146
Data columns (total 5 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   INDICADOR  852 non-null    object 
 1   PAIS       852 non-null    object 
 2   ANO        852 non-null    int64  
 3   VALOR      852 non-null    float64
 4   UNIDADE    852 non-null    object 
dtypes: float64(1), int64(1), object(3)
memory usage: 39.9+ KB


In [84]:
nacoesunidasfinal = pd.concat([nacoesunidas_2, nacoes_poverty], ignore_index=True)

In [85]:
nacoesunidasfinal.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 312532 entries, 0 to 312531
Data columns (total 5 columns):
 #   Column     Non-Null Count   Dtype 
---  ------     --------------   ----- 
 0   INDICADOR  312532 non-null  object
 1   PAIS       312532 non-null  object
 2   ANO        312532 non-null  int64 
 3   VALOR      309525 non-null  object
 4   UNIDADE    312532 non-null  object
dtypes: int64(1), object(4)
memory usage: 11.9+ MB


In [86]:
nacoesunidasfinal

INDICADOR    PAIS   ANO   
0       Foreign direct investment, net inflows, as a p...   World  1970  \
1       Foreign direct investment, net inflows, as a p...   World  1971   
2       Foreign direct investment, net inflows, as a p...   World  1972   
3       Foreign direct investment, net inflows, as a p...   World  1973   
4       Foreign direct investment, net inflows, as a p...   World  1974   
...                                                   ...     ...   ...   
312527             Poverty gap ratio at USD 1 a day (PPP)  Zambia  1998   
312528             Poverty gap ratio at USD 1 a day (PPP)  Zambia  2003   
312529             Poverty gap ratio at USD 1 a day (PPP)  Zambia  2004   
312530             Poverty gap ratio at USD 1 a day (PPP)  Zambia  2006   
312531             Poverty gap ratio at USD 1 a day (PPP)  Zambia  2010   

          VALOR  UNIDADE  
0       0.47921       PT  
1        0.4702       PT  
2       0.42351       PT  
3       0.47988       PT  
4       0.49684       PT  
...         ...      ...  
312527    26.94  PERCENT  
312528    27.13  PERCENT  
312529    32.76  PERCENT  
312530    37.02  PERCENT  
312531    41.78  PERCENT  

[312532 rows x 5 columns]

In [87]:
# Substituir os valores da coluna 'PAIS' pelos nomes dos países
nacoesunidasfinal['PAIS'] = nacoesunidasfinal['PAIS'].replace({1:'World',1:'World',1:'World',100:'Bulgaria',100:'Bulgaria',104:'Myanmar',104:'Myanmar',108:'Burundi',108:'Burundi',11:'Western Africa',11:'Western Africa',112:'Belarus',112:'Belarus',116:'Cambodia',116:'Cambodia',12:'Algeria',12:'Algeria',120:'Cameroon',120:'Cameroon',124:'Canada',124:'Canada',13:'Central America',13:'Central America',132:'Cabo Verde',132:'Cabo Verde',136:'Cayman Islands',136:'Cayman Islands',14:'Eastern Africa',14:'Eastern Africa',140:'Central African Republic',140:'Central African Republic',142:'Asia',142:'Asia',143:'Central Asia',143:'Central Asia',144:'Sri Lanka',144:'Sri Lanka',145:'Western Asia',145:'Western Asia',148:'Chad',148:'Chad',15:'Northern Africa',15:'Northern Africa',150:'Europe',150:'Europe',151:'Eastern Europe',151:'Eastern Europe',152:'Chile',152:'Chile',154:'Northern Europe',154:'Northern Europe',155:'Western Europe',155:'Western Europe',156:'China',156:'China',17:'Middle Africa',17:'Middle Africa',170:'Colombia',170:'Colombia',174:'Comoros',174:'Comoros',178:'Democratic Republic of the Congo',178:'Democratic Republic of the Congo',18:'Southern Africa',18:'Southern Africa',180:'Democratic Republic of the Congo',180:'Democratic Republic of the Congo',188:'Costa Rica',188:'Costa Rica',19:'Americas',19:'Americas',191:'Croatia',191:'Croatia',196:'Cyprus',196:'Cyprus',199:'Least Developed Countries ',199:'Least Developed Countries ',2:'Africa',2:'Africa',202:'Sub',202:'Sub',203:'Czechia',203:'Czechia',204:'Benin',204:'Benin',208:'Denmark',208:'Denmark',21:'Northern America',21:'Northern America',212:'Dominica',212:'Dominica',214:'Dominican Republic',214:'Dominican Republic',218:'Ecuador',218:'Ecuador',222:'El Salvador',222:'El Salvador',226:'Equatorial Guinea',226:'Equatorial Guinea',231:'Ethiopia',231:'Ethiopia',232:'Eritrea',232:'Eritrea',233:'Estonia',233:'Estonia',24:'Angola',24:'Angola',242:'Fiji',242:'Fiji',246:'Finland',246:'Finland',250:'France',250:'France',258:'French Polynesia',258:'French Polynesia',262:'Djibouti',262:'Djibouti',266:'Gabon',266:'Gabon',268:'Georgia',268:'Georgia',270:'Gambia',270:'Gambia',275:'State of Palestine',275:'State of Palestine',276:'Germany',276:'Germany',28:'Antigua and Barbuda',28:'Antigua and Barbuda',288:'Ghana',288:'Ghana',296:'Kiribati',296:'Kiribati',30:'Eastern Asia',30:'Eastern Asia',300:'Greece',300:'Greece',308:'Grenada',308:'Grenada',31:'Azerbaijan',31:'Azerbaijan',32:'Argentina',32:'Argentina',320:'Guatemala',320:'Guatemala',324:'Guinea',324:'Guinea',328:'Guyana',328:'Guyana',332:'Haiti',332:'Haiti',34:'Southern Asia',34:'Southern Asia',340:'Honduras',340:'Honduras',344:'China Hong Kong',344:'China Hong Kong',348:'Hungary',348:'Hungary',35:'South',35:'South',352:'Iceland',352:'Iceland',356:'India',356:'India',36:'Australia',36:'Australia',360:'Indonesia',360:'Indonesia',364:'Iran ',364:'Iran ',368:'Iraq',368:'Iraq',372:'Ireland',372:'Ireland',376:'Israel',376:'Israel',380:'Italy',380:'Italy',384:'C',384:'C',388:'Jamaica',388:'Jamaica',39:'Southern Europe',39:'Southern Europe',392:'Japan',392:'Japan',398:'Kazakhstan',398:'Kazakhstan',4:'Afghanistan',4:'Afghanistan',40:'Austria',40:'Austria',400:'Jordan',400:'Jordan',404:'Kenya',404:'Kenya',410:'Republic of Korea',410:'Republic of Korea',412:'Kosovo',412:'Kosovo',414:'Kuwait',414:'Kuwait',417:'Kyrgyzstan',417:'Kyrgyzstan',418:'Lao People',418:'Lao People',419:'Latin America and the Caribbean',419:'Latin America and the Caribbean',422:'Lebanon',422:'Lebanon',426:'Lesotho',426:'Lesotho',428:'Latvia',428:'Latvia',430:'Liberia',430:'Liberia',432:'Landlocked developing countries ',432:'Landlocked developing countries ',434:'Libya',434:'Libya',438:'Liechtenstein',438:'Liechtenstein',44:'Bahamas',44:'Bahamas',440:'Lithuania',440:'Lithuania',442:'Luxembourg',442:'Luxembourg',446:'China Macao',446:'China Macao',450:'Madagascar',450:'Madagascar',454:'Malawi',454:'Malawi',458:'Malaysia',458:'Malaysia',462:'Maldives',462:'Maldives',466:'Mali',466:'Mali',470:'Malta',470:'Malta',478:'Mauritania',478:'Mauritania',48:'Bahrain',48:'Bahrain',480:'Mauritius',480:'Mauritius',484:'Mexico',484:'Mexico',496:'Mongolia',496:'Mongolia',498:'Republic of Moldova',498:'Republic of Moldova',499:'Montenegro',499:'Montenegro',5:'South America',5:'South America',50:'Bangladesh',50:'Bangladesh',504:'Morocco',504:'Morocco',508:'Mozambique',508:'Mozambique',51:'Armenia',51:'Armenia',512:'Oman',512:'Oman',513:'Europe and Northern America',513:'Europe and Northern America',516:'Namibia',516:'Namibia',52:'Barbados',52:'Barbados',524:'Nepal',524:'Nepal',528:'Netherlands ',528:'Netherlands ',53:'Australia and New Zealand',53:'Australia and New Zealand',531:'Cura',531:'Cura',533:'Aruba',533:'Aruba',534:'Sint Maarten ',534:'Sint Maarten ',54:'Melanesia',54:'Melanesia',540:'New Caledonia',540:'New Caledonia',543:'Oceania ',543:'Oceania ',548:'Vanuatu',548:'Vanuatu',554:'New Zealand',554:'New Zealand',558:'Nicaragua',558:'Nicaragua',56:'Belgium',56:'Belgium',562:'Niger',562:'Niger',566:'Nigeria',566:'Nigeria',57:'Micronesia',57:'Micronesia',578:'Norway',578:'Norway',583:'Micronesia ',583:'Micronesia ',584:'Marshall Islands',584:'Marshall Islands',585:'Palau',585:'Palau',586:'Pakistan',586:'Pakistan',591:'Panama',591:'Panama',598:'Papua New Guinea',598:'Papua New Guinea',60:'Bermuda',60:'Bermuda',600:'Paraguay',600:'Paraguay',604:'Peru',604:'Peru',608:'Philippines',608:'Philippines',61:'Polynesia',61:'Polynesia',616:'Poland',616:'Poland',62:'Central and Southern Asia',62:'Central and Southern Asia',620:'Portugal',620:'Portugal',624:'Guinea',624:'Guinea',626:'Timor',626:'Timor',634:'Qatar',634:'Qatar',64:'Bhutan',64:'Bhutan',642:'Romania',642:'Romania',643:'Russian Federation',643:'Russian Federation',646:'Rwanda',646:'Rwanda',659:'Saint Kitts and Nevis',659:'Saint Kitts and Nevis',662:'Saint Lucia',662:'Saint Lucia',670:'Saint Vincent and the Grenadines',670:'Saint Vincent and the Grenadines',678:'Sao Tome and Principe',678:'Sao Tome and Principe',68:'Bolivia ',68:'Bolivia ',682:'Saudi Arabia',682:'Saudi Arabia',686:'Senegal',686:'Senegal',688:'Serbia',688:'Serbia',690:'Seychelles',690:'Seychelles',694:'Sierra Leone',694:'Sierra Leone',70:'Bosnia and Herzegovina',70:'Bosnia and Herzegovina',702:'Singapore',702:'Singapore',703:'Slovakia',703:'Slovakia',704:'Viet Nam',704:'Viet Nam',705:'Slovenia',705:'Slovenia',706:'Somalia',706:'Somalia',710:'South Africa',710:'South Africa',716:'Zimbabwe',716:'Zimbabwe',72:'Botswana',72:'Botswana',722:'Small island developing States ',722:'Small island developing States ',724:'Spain',724:'Spain',728:'South Sudan',728:'South Sudan',729:'Sudan',729:'Sudan',740:'Suriname',740:'Suriname',747:'Northern Africa and Western Asia',747:'Northern Africa and Western Asia',748:'Eswatini',748:'Eswatini',752:'Sweden',752:'Sweden',753:'Eastern and South',753:'Eastern and South',756:'Switzerland',756:'Switzerland',76:'Brazil',76:'Brazil',760:'Syrian Arab Republic',760:'Syrian Arab Republic',762:'Tajikistan',762:'Tajikistan',764:'Thailand',764:'Thailand',768:'Togo',768:'Togo',776:'Tonga',776:'Tonga',780:'Trinidad and Tobago',780:'Trinidad and Tobago',784:'United Arab Emirates',784:'United Arab Emirates',788:'Tunisia',788:'Tunisia',792:'T',792:'T',795:'Turkmenistan',795:'Turkmenistan',796:'Turks and Caicos Islands',796:'Turks and Caicos Islands',798:'Tuvalu',798:'Tuvalu',8:'Albania',8:'Albania',800:'Uganda',800:'Uganda',804:'Ukraine',804:'Ukraine',807:'North Macedonia',807:'North Macedonia',818:'Egypt',818:'Egypt',826:'United Kingdom of Great Britain and Northern Ireland',826:'United Kingdom of Great Britain and Northern Ireland',834:'United Republic of Tanzania',834:'United Republic of Tanzania',84:'Belize',84:'Belize',840:'United States of America',840:'United States of America',854:'Burkina Faso',854:'Burkina Faso',858:'Uruguay',858:'Uruguay',860:'Uzbekistan',860:'Uzbekistan',862:'Venezuela ',862:'Venezuela ',882:'Samoa',882:'Samoa',887:'Yemen',887:'Yemen',894:'Zambia',894:'Zambia',9:'Oceania',9:'Oceania',90:'Solomon Islands',90:'Solomon Islands',96:'Brunei Darussalam',96:'Brunei Darussalam',1:'World',2:'Africa',15:'Northern Africa',12:'Algeria',434:'Libya',504:'Morocco',729:'Sudan',732:'Western Sahara',788:'Tunisia',818:'Egypt',202:'Sub',11:'Western Africa',132:'Cabo Verde',178:'Democratic Republic of the Congo',204:'Benin',270:'Gambia',288:'Ghana',324:'Guinea',384:'C',430:'Liberia',466:'Mali',478:'Mauritania',562:'Niger',566:'Nigeria',624:'Guinea',654:'Saint Helena',686:'Senegal',694:'Sierra Leone',768:'Togo',854:'Burkina Faso',14:'Eastern Africa',86:'British Indian Ocean Territory',108:'Burundi',174:'Comoros',175:'Mayotte',231:'Ethiopia',232:'Eritrea',260:'French Southern Territories',262:'Djibouti',404:'Kenya',450:'Madagascar',454:'Malawi',480:'Mauritius',508:'Mozambique',638:'R',646:'Rwanda',690:'Seychelles',706:'Somalia',716:'Zimbabwe',728:'South Sudan',736:'Sudan ',800:'Uganda',834:'United Republic of Tanzania',894:'Zambia',17:'Middle Africa',24:'Angola',120:'Cameroon',140:'Central African Republic',148:'Chad',180:'Democratic Republic of the Congo',226:'Equatorial Guinea',266:'Gabon',678:'Sao Tome and Principe',18:'Southern Africa',72:'Botswana',426:'Lesotho',516:'Namibia',710:'South Africa',748:'Eswatini',9:'Oceania',53:'Australia and New Zealand',36:'Australia',162:'Christmas Island',166:'Cocos ',334:'Heard Island and McDonald Islands',554:'New Zealand',574:'Norfolk Island',54:'Melanesia',90:'Solomon Islands',242:'Fiji',540:'New Caledonia',548:'Vanuatu',598:'Papua New Guinea',57:'Micronesia',296:'Kiribati',316:'Guam',520:'Nauru',580:'Northern Mariana Islands',581:'United States Minor Outlying Islands',583:'Micronesia ',584:'Marshall Islands',585:'Palau',61:'Polynesia',16:'American Samoa',184:'Cook Islands',258:'French Polynesia',570:'Niue',612:'Pitcairn',772:'Tokelau',776:'Tonga',798:'Tuvalu',876:'Wallis and Futuna Islands',882:'Samoa',543:'Oceania ',10:'Antarctica',19:'Americas',21:'Northern America',60:'Bermuda',124:'Canada',304:'Greenland',666:'Saint Pierre and Miquelon',840:'United States of America',419:'Latin America and the Caribbean',5:'South America',32:'Argentina',68:'Bolivia ',74:'Bouvet Island',76:'Brazil',152:'Chile',170:'Colombia',218:'Ecuador',238:'Falkland Islands ',239:'South Georgia and the South Sandwich Islands',254:'French Guiana',328:'Guyana',600:'Paraguay',604:'Peru',740:'Suriname',858:'Uruguay',862:'Venezuela ',13:'Central America',84:'Belize',188:'Costa Rica',222:'El Salvador',320:'Guatemala',340:'Honduras',484:'Mexico',558:'Nicaragua',591:'Panama',29:'Caribbean',28:'Antigua and Barbuda',44:'Bahamas',52:'Barbados',92:'British Virgin Islands',136:'Cayman Islands',192:'Cuba',212:'Dominica',214:'Dominican Republic',308:'Grenada',312:'Guadeloupe',332:'Haiti',388:'Jamaica',474:'Martinique',500:'Montserrat',530:'Netherlands Antilles',531:'Cura',533:'Aruba',534:'Sint Maarten ',535:'Bonaire',630:'Puerto Rico',652:'Saint Barth',659:'Saint Kitts and Nevis',660:'Anguilla',662:'Saint Lucia',663:'Saint Martin ',670:'Saint Vincent and the Grenadines',780:'Trinidad and Tobago',796:'Turks and Caicos Islands',850:'United States Virgin Islands',62:'Central and Southern Asia',142:'Asia',30:'Eastern Asia',156:'China',158:'Other non',344:'China Hong Kong',392:'Japan',408:'Democratic People',410:'Republic of Korea',446:'China Macao',496:'Mongolia',34:'Southern Asia',4:'Afghanistan',50:'Bangladesh',64:'Bhutan',144:'Sri Lanka',356:'India',364:'Iran ',462:'Maldives',524:'Nepal',586:'Pakistan',35:'South',96:'Brunei Darussalam',104:'Myanmar',116:'Cambodia',360:'Indonesia',418:'Lao People',458:'Malaysia',608:'Philippines',626:'Timor',702:'Singapore',704:'Viet Nam',764:'Thailand',143:'Central Asia',398:'Kazakhstan',417:'Kyrgyzstan',762:'Tajikistan',795:'Turkmenistan',860:'Uzbekistan',145:'Western Asia',31:'Azerbaijan',48:'Bahrain',51:'Armenia',196:'Cyprus',268:'Georgia',275:'State of Palestine',368:'Iraq',369:'Iraq ',370:'Iraq ',376:'Israel',400:'Jordan',414:'Kuwait',422:'Lebanon',512:'Oman',634:'Qatar',682:'Saudi Arabia',760:'Syrian Arab Republic',784:'United Arab Emirates',792:'T',887:'Yemen',223:'Eastern Asia ',150:'Europe',39:'Southern Europe',8:'Albania',20:'Andorra',70:'Bosnia and Herzegovina',191:'Croatia',292:'Gibraltar',300:'Greece',336:'Holy See',380:'Italy',470:'Malta',499:'Montenegro',620:'Portugal',674:'San Marino',688:'Serbia',705:'Slovenia',724:'Spain',807:'North Macedonia',891:'Serbia and Montenegro ',151:'Eastern Europe',100:'Bulgaria',112:'Belarus',203:'Czechia',348:'Hungary',412:'Kosovo',498:'Republic of Moldova',616:'Poland',642:'Romania',643:'Russian Federation',703:'Slovakia',804:'Ukraine',154:'Northern Europe',208:'Denmark',233:'Estonia',234:'Faroe Islands',246:'Finland',352:'Iceland',372:'Ireland',428:'Latvia',440:'Lithuania',578:'Norway',744:'Svalbard and Jan Mayen Islands',752:'Sweden',826:'United Kingdom of Great Britain and Northern Ireland',827:'United Kingdom ',828:'United Kingdom ',829:'United Kingdom ',830:'Channel Islands',680:'Sark',831:'Guernsey',832:'Jersey',833:'Isle of Man',155:'Western Europe',40:'Austria',56:'Belgium',250:'France',276:'Germany',438:'Liechtenstein',442:'Luxembourg',492:'Monaco',528:'Netherlands ',756:'Switzerland',513:'Europe and Northern America',747:'Northern Africa and Western Asia',753:'Eastern and South',127:'Southern Asia ',135:'Caucasus and Central Asia',199:'Least Developed Countries ',420:'Latin America',432:'Landlocked developing countries ',485:'Western Asia ',514:'Developed regions ',515:'Developing regions',518:'Eastern Asia ',593:'Development Assistance Committee members ',722:'Small island developing States ',738:'Sub',746:'Northern Africa ',890:'Yugoslavia ',901:'Africa ',903:'Central and Eastern Europe ',904:'Middle East and North Africa ',905:'Middle East ',911:'Low income economies ',921:'ODA residual',97:'European Union',487:'Other non',527:'Other non',636:'Other non',906:'North America ',917:'International Centers ',918:'European Union ',919:'Regional Centres ',951:'Other Africa ',952:'Unallocated by country',99018:'FAO Major Fishing Area',99019:'FAO Major Fishing Area',99020:'FAO Major Fishing Area',99021:'FAO Major Fishing Area',99022:'FAO Major Fishing Area',99023:'FAO Major Fishing Area',99024:'FAO Major Fishing Area',99025:'FAO Major Fishing Area',99026:'FAO Major Fishing Area',99027:'FAO Major Fishing Area',99028:'FAO Major Fishing Area',99029:'FAO Major Fishing Area',99030:'FAO Major Fishing Area',99031:'FAO Major Fishing Area',99032:'FAO Major Fishing Area',99035:'Residual',99036:'Residual',99037:'Residual',99038:'Residual',99039:'Residual',99040:'Residual',99041:'Belgium and Luxembourg',99049:'Residual',99050:'Residual'})

In [88]:
nacoesunidasfinal['UNIDADE'].unique()

array(['PT', 'RO', 'NUMBER', 'LITRES_PURE_ALCOHOL',
       'PER_1000_LIVE_BIRTHS', 'PER_10000_POP', 'PER_100000_POP',
       'CUR_LCU', 'PERCENT'], dtype=object)

In [89]:
nacoesunidasfinal['UNIDADE'] = nacoesunidasfinal['UNIDADE'].replace({'PT':'PERCENT','RO':'RATIO'})

In [90]:
nacoesunidasfinal['INDICADOR'] = nacoesunidasfinal['INDICADOR'].replace({'SI_POV_DAY1':'Proportion of population below international poverty line',
    'SI_POV_EMP1':'Employed population below international poverty line','SE_ACS_ELECT':'Proportion of schools with access to electricity, by education level [4.a.1]',
    'SH_DYN_IMRTN':'Infant deaths (number) [3.2.1]',
    'SH_FPL_MTMM':'Proportion of women married or in a union of reproductive age 15-49 who have their need for family planning satisfied with modern methods [3.7.1]'
})

In [91]:
nacoesunidasfinal['UNIDADE'].unique()

array(['PERCENT', 'RATIO', 'NUMBER', 'LITRES_PURE_ALCOHOL',
       'PER_1000_LIVE_BIRTHS', 'PER_10000_POP', 'PER_100000_POP',
       'CUR_LCU'], dtype=object)

In [92]:
nacoesunidasfinal['INDICADOR'].unique()

array(['Foreign direct investment, net inflows, as a proportion of GDP (%)',
       'Renewable energy share in the total final energy consumption',
       'Annual inflation, consumer prices (%)',
       'Proportion of individuals who own a mobile telephone',
       'Annual GDP growth (%)',
       'Annual growth rate of real GDP per capita',
       'Proportion of schools with access to electricity, by education level [4.a.1]',
       'Proportion of youth and adults with information and communications technology (ICT) skills',
       'Participation rate in formal and non-formal education and training',
       'Gender parity index for youth/adults with information and communications technology (ICT) skills',
       'Adjusted gender parity index for participation rate in formal and non-formal education and training (ratio)',
       'Proportion of elected seats held by women in deliberative bodies of local government',
       'Proportion of seats held by women in national parliaments (% of 

In [93]:
nacoesunidasfinal.loc[nacoesunidasfinal['INDICADOR'] == 'Proportion of population below international poverty line']

INDICADOR    PAIS   ANO   
272625  Proportion of population below international p...   World  1981  \
272626  Proportion of population below international p...   World  1982   
272627  Proportion of population below international p...   World  1983   
272628  Proportion of population below international p...   World  1984   
272629  Proportion of population below international p...   World  1985   
...                                                   ...     ...   ...   
282395  Proportion of population below international p...  Zambia  2015   
282396  Proportion of population below international p...  Zambia  2015   
282397  Proportion of population below international p...  Zambia  2015   
282398  Proportion of population below international p...  Zambia  2015   
282399  Proportion of population below international p...  Zambia  2015   

          VALOR  UNIDADE  
272625     43.6  PERCENT  
272626     43.2  PERCENT  
272627     42.5  PERCENT  
272628     41.1  PERCENT  
272629     39.7  PERCENT  
...         ...      ...  
282395  686.308  PERCENT  
282396  678.016  PERCENT  
282397  573.037  PERCENT  
282398  612.914  PERCENT  
282399  614.164  PERCENT  

[9775 rows x 5 columns]

In [94]:
nacoesunidasfinal.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 312532 entries, 0 to 312531
Data columns (total 5 columns):
 #   Column     Non-Null Count   Dtype 
---  ------     --------------   ----- 
 0   INDICADOR  312532 non-null  object
 1   PAIS       312532 non-null  object
 2   ANO        312532 non-null  int64 
 3   VALOR      309525 non-null  object
 4   UNIDADE    312532 non-null  object
dtypes: int64(1), object(4)
memory usage: 11.9+ MB


In [272]:
nacoesunidasfinal = nacoesunidasfinal[nacoesunidasfinal['INDICADOR'].isin(['Proportion of individuals who own a mobile telephone',
    'Annual GDP growth (%)',
    'Annual growth rate of real GDP per capita',
    'Number of seats held by women in national parliaments',
    'Current number of seats in national parliaments',
    'Health worker density, by type of occupation (per 10,000 population)',
    'Health worker distribution, by sex and type of occupation',
    'Prevalence of undernourishment',
    'Proportion of population using basic drinking water services',
    'Proportion of population using basic sanitation services',
    'Proportion of population below international poverty line',
    'Poverty gap ratio at USD 1 a day (PPP)'])]

In [273]:
nacoesunidasfinal['VALOR'] = nacoesunidasfinal['VALOR'].astype(float)

C:\Users\carlo\AppData\Local\Temp\ipykernel_16972\3772665501.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  nacoesunidasfinal['VALOR'] = nacoesunidasfinal['VALOR'].astype(float)


In [ ]:
# dados_desproporcionais = variavel_prov.loc[(variavel_prov['INDICADOR'] == 'Proportion of population below international poverty line') &
#                                              (variavel_prov['VALOR'] > 0)]
#                                              
# dados_desproporcionais

-- Localizar e eliminar os valores discrepantes do dataframe
-- Criar um novo dataframe para armazenar os registros únicos
df_unique = pd.DataFrame()

-- Iterar por cada grupo de país e ano
for (pais, ano), group in dados_desproporcionais.groupby(['PAIS', 'ANO']):
    -- Selecionar a linha com menor valor
    min_row = group.loc[group['VALOR'].idxmin()]
    -- Adicionar a linha ao dataframe de registros únicos
    df_unique = pd.concat([df_unique, min_row], axis=1)
-- Exibir o novo dataframe com os registros únicos
dados_desproporcionais = df_unique  -- Transpor o dataframe para ajustar a orientação
dados_desproporcionais = dados_desproporcionais.T  -- Exibir o dataframe
dados_desproporcionais

## dados_desproporcionais.info()

Index: 9104 entries, 272625 to 282399
Data columns (total 5 columns):
     Column     Non-Null Count  Dtype   |
---  ------     --------------  -----   |
 0   INDICADOR  9104 non-null   object  |
 1   PAIS       9104 non-null   object  |
 2   ANO        9104 non-null   int64   |
 3   VALOR      9104 non-null   float64 |
 4   UNIDADE    9104 non-null   object  |
dtypes: float64(1), int64(1), object(3) |
memory usage: 426.8+ KB                 |

# dados_desproporcionais['PAIS'].info()

Index: 9104 entries, 272625 to 282399
Data columns (total 5 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   INDICADOR  9104 non-null   object 
 1   PAIS       9104 non-null   object 
 2   ANO        9104 non-null   int64  
 3   VALOR      9104 non-null   float64
 4   UNIDADE    9104 non-null   object 
dtypes: float64(1), int64(1), object(3)
memory usage: 426.8+ KB

In [277]:
nacoesunidasfinal.info()

<class 'pandas.core.frame.DataFrame'>
Index: 80022 entries, 25625 to 312531
Data columns (total 5 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   INDICADOR  80022 non-null  object 
 1   PAIS       80022 non-null  object 
 2   ANO        80022 non-null  int64  
 3   VALOR      77121 non-null  float64
 4   UNIDADE    80022 non-null  object 
dtypes: float64(1), int64(1), object(3)
memory usage: 3.7+ MB


In [278]:
# Eliminar as regiões e sub-regiões para ficar somente com os países
nacoesunidasfinal = nacoesunidasfinal[nacoesunidasfinal['PAIS'].isin(['Albania','Algeria','Angola','Caribbean','Azerbaijan','Argentina','Australia','Austria','Bangladesh','Armenia','Australia and New Zealand','Melanesia','Belgium','Bhutan','Bolivia ','Bosnia and Herzegovina','Botswana','Brazil','Belize','Solomon Islands','Bulgaria','Myanmar','Burundi','Belarus','Cameroon','Canada','Cabo Verde','Caucasus and Central Asia','Sri Lanka','Chad','Chile','China','Colombia','Comoros','Congo','Democratic Republic of the Congo','Costa Rica','Croatia','Cyprus','Czechia','Benin','Denmark','Dominican Republic','Ecuador','El Salvador','Ethiopia','Estonia','Fiji','Finland','France','Djibouti','Gabon','Georgia','Gambia','State of Palestine','Germany','Ghana','Kiribati','Greece','Guatemala','Guinea','Guyana','Haiti','Honduras','Hungary','Iceland','India','Indonesia','Iran ','Iraq','Ireland','Israel','Italy','Jamaica','Japan','Kazakhstan','Jordan','Kenya','Republic of Korea','Kyrgyzstan','Lao People','Lesotho','Latvia','Liberia','Lithuania','Luxembourg','Madagascar','Malawi','Malaysia','Maldives','Mali','Malta','Mauritania','Mauritius','Mexico','Mongolia','Republic of Moldova','Montenegro','Morocco','Mozambique','Namibia','Nauru','Nepal','Netherlands ','Vanuatu','Nicaragua','Niger','Nigeria','Norway','Micronesia ','Marshall Islands','Pakistan','Panama','Papua New Guinea','Paraguay','Peru','Philippines','Poland','Portugal','Timor','Romania','Russian Federation','Rwanda','Saint Lucia','Sao Tome and Principe','Senegal','Serbia','Seychelles','Sierra Leone','Slovakia','Viet Nam','Slovenia','South Africa','Zimbabwe','Spain','Sudan','Suriname','Eswatini','Sweden','Switzerland','Syrian Arab Republic','Tajikistan','Thailand','Togo','Tonga','Trinidad and Tobago','Tunisia','Turkmenistan','Tuvalu','Uganda','Ukraine','North Macedonia','Egypt','United Kingdom of Great Britain and Northern Ireland','United Republic of Tanzania','United States of America','Burkina Faso','Uruguay','Uzbekistan','Venezuela ','Samoa','Yemen','Zambia'])]
nacoesunidasfinal.info()

<class 'pandas.core.frame.DataFrame'>
Index: 57723 entries, 25637 to 312531
Data columns (total 5 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   INDICADOR  57723 non-null  object 
 1   PAIS       57723 non-null  object 
 2   ANO        57723 non-null  int64  
 3   VALOR      56428 non-null  float64
 4   UNIDADE    57723 non-null  object 
dtypes: float64(1), int64(1), object(3)
memory usage: 2.6+ MB


In [ ]:
# Localizar e eliminar os valores discrepantes do dataframe
# Criar um novo dataframe para armazenar os registros únicos
df_unique = pd.DataFrame()

# Define uma variável provisória para armazenar os dados desproporcionais a iterar
variavel_prov = nacoesunidasfinal
# Iterar por cada grupo de país e ano
for (pais, ano), group in dados_desproporcionais.groupby(['PAIS', 'ANO']):
    # Selecionar a linha com menor valor
    min_row = group.loc[group['VALOR'].idxmin()]
    # Adicionar a linha ao dataframe de registros únicos
    df_unique = pd.concat([df_unique, min_row], axis=1)
# Exibir o novo dataframe com os registros únicos
dados_desproporcionais = df_unique  # Transpor o dataframe para ajustar a orientação
dados_desproporcionais = dados_desproporcionais.T  # Exibir o dataframe
dados_desproporcionais

In [50]:
ONU_Unificado = pd.pivot_table(variavel_prov, index=['PAIS','ANO'], columns='INDICADOR', values='VALOR', aggfunc='mean')

In [51]:
ONU_Unificado

INDICADOR         Annual GDP growth (%)   
PAIS        ANO                           
Afghanistan 2000                    NaN  \
            2001                    NaN   
            2002                    NaN   
            2003                883.228   
            2004                141.412   
...                                 ...   
Zimbabwe    2019               -633.245   
            2020               -781.695   
            2021                846.802   
            2022                    NaN   
            2023                    NaN   

INDICADOR         Annual growth rate of real GDP per capita   
PAIS        ANO                                               
Afghanistan 2000                                      -6.71  \
            2001                                      -6.33   
            2002                                      39.29   
            2003                                       7.40   
            2004                                      -1.11   
...                                                     ...   
Zimbabwe    2019                                      -7.95   
            2020                                      -7.20   
            2021                                       4.15   
            2022                                        NaN   
            2023                                        NaN   

INDICADOR         Current number of seats in national parliaments   
PAIS        ANO                                                     
Afghanistan 2000                                              NaN  \
            2001                                              NaN   
            2002                                              NaN   
            2003                                              NaN   
            2004                                              NaN   
...                                                           ...   
Zimbabwe    2019                                              NaN   
            2020                                              NaN   
            2021                                            270.0   
            2022                                            265.0   
            2023                                            265.0   

INDICADOR         Health worker density, by type of occupation (per 10,000 population)   
PAIS        ANO                                                                          
Afghanistan 2000                                                NaN                     \
            2001                                             1.1750                      
            2002                                                NaN                      
            2003                                                NaN                      
            2004                                                NaN                      
...                                                             ...                      
Zimbabwe    2019                                             0.1600                      
            2020                                             5.8375                      
            2021                                                NaN                      
            2022                                                NaN                      
            2023                                                NaN                      

INDICADOR         Health worker distribution, by sex and type of occupation   
PAIS        ANO                                                               
Afghanistan 2000                                                NaN          \
            2001                                                NaN           
            2002                                                NaN           
            2003                                                NaN           
            2004                                                NaN          

In [68]:
ONU_Unificado.to_excel('ONU_Unificado.xlsx')

In [89]:
ONU_Unificado['Proportion of population below international poverty line']

PAIS         ANO 
Afghanistan  2000        NaN
             2001        NaN
             2002        NaN
             2003        NaN
             2004        NaN
                      ...   
Zimbabwe     2019    327.988
             2020        NaN
             2021        NaN
             2022        NaN
             2023        NaN
Name: Proportion of population below international poverty line, Length: 13323, dtype: float64

In [46]:
# Lista de indicadores
indicadores = [
    'Proportion of individuals who own a mobile telephone',
    'Annual GDP growth (%)',
    'Annual growth rate of real GDP per capita',
    'Number of seats held by women in national parliaments',
    'Current number of seats in national parliaments',
    'Health worker density, by type of occupation (per 10,000 population)',
    'Health worker distribution, by sex and type of occupation',
    'Prevalence of undernourishment',
    'Proportion of population using basic drinking water services',
    'Proportion of population using basic sanitation services',
    'Proportion of population below international poverty line',
    'Poverty gap ratio at USD 1 a day (PPP)'
]

# Iterar sobre os indicadores
for indicador in indicadores:
    # Filtrar os dados pelo indicador atual
    df_filtrado = nacoesunidasfinal[nacoesunidasfinal['INDICADOR'] == indicador]
    
    # Nome do arquivo CSV
    nome_arquivo = f"{indicador}.csv"
    
    # Salvar o DataFrame filtrado como CSV
    df_filtrado.to_csv(nome_arquivo, index=False)


In [ ]:
nacoesunidasfinal